---
title: Microbial community assembly in a multi-layer dendritic metacommunity
author: "Nathan I. Wisnoski and Jay T. Lennon"
date: "2019-01-04"
output: pdf_document
---

# Initial setup 
```{r, include=FALSE}
knitr::opts_chunk$set(
  echo = TRUE, 
  fig.align = "center", 
  fig.width = 6,
  message = FALSE,
  warning = FALSE
)


library("adespatial")
library("pander")
library("stringr")
library("picante")
library("phytools")
library("tidyverse")
library("cowplot")
library("ggrepel")
library("progress")
library("doSNOW")
library("vegan")
library("colorspace")

theme_set(theme_cowplot() +
            theme(axis.title = element_text(size = 12),
                  axis.text = element_text(size = 10),
                  legend.position = "top", 
                  strip.text = element_text(size = 12),
                  strip.background = element_blank(),
                  legend.text = element_text(size = 10),
                  legend.title = element_text(size = 12)))

# used https://www.color-hex.com/color-palette/71977
my.colors <- c("#d5ba94", "#3a4f6a")

# load external functions
source("analysis/mothur_tools.R")
source("analysis/hja_functions.R")
source("analysis/comdistnt.par.R")
source("analysis/ses.comdistnt2.R")

```

First, we load the data. This includes the site-by-species matrix (generated in Mothur, v. 1.41.1), the RDP taxonomy, the environmental data, and the phylogenetic tree (generated with FastTreeMP).
```{r, echo=FALSE, message=FALSE}
## Import Shared, Design, and Environment Files

# Define Inputs
# Design = general design file for experiment
# shared = OTU table from mothur with sequence similarity clustering
# Taxonomy = Taxonomic information for each OTU

# Import Design
design.total <- read.delim("data/design.txt", header=T, row.names=1)

# Import Shared Files
OTUs <- read.otu(shared = "data/hja_streams.shared", cutoff = "0.03") # 97% Similarity

# Import Taxonomy
OTU.tax <- read.tax(taxonomy = "data/hja_streams.0.03.taxonomy", format = "rdp")

# Import Env
options(readr.default_locale=readr::locale(tz="America/Indiana/Indianapolis"))
env.total <- read_csv("data/hja_env.csv", 
                      col_types = cols(
                        date = col_date(format = "%d-%b-%y"),
                        habitat = col_factor(levels = c("sediment", "water"), ordered = F)))

# Check to see that the OTU table rows are same order as design and then rename OTU table rows 
sites <- as.data.frame(str_split(rownames(design.total), "_", simplify = T))
colnames(sites) <- c("ws", "site", "habitat")
oturows <- rownames(OTUs) %>% str_sub(7)
if(all.equal.character(paste0(sites$ws, sites$site, sites$habitat), oturows)){
  rownames(OTUs) <- rownames(design.total)
}
```

Next, we will clean up the data. I'll remove any sample that didn't get 10000 reads. Then also cut those samples from the environment and design tables.
```{r}
# Sequencing Coverage
coverage <- rowSums(OTUs)

# Remove Low Coverage Samples
cutoff <- 10000
lows <- which(coverage < cutoff)
OTUs <- OTUs[-which(coverage < cutoff), ]
design <- design.total[-which(coverage < cutoff), ]
env <- env.total[-which(coverage < cutoff), ]

# Remove OTUs with less than 5 occurances across all sites
OTUs <- OTUs[, which(colSums(OTUs) > 2)]

OTUs <- OTUs[-which(env$sample == "W1_20_W"),]
design <- design[-which(env$sample == "W1_20_W"),]
env <- env[-which(env$sample == "W1_20_W"),]
```

Here, I'll read in the dendritic distances and add a tiny bit of jitter to the spatial distances so nearby sites aren't identical. Then, I'll calculate the earth distance in meters.
```{r}
den.dists <- make.dendritic.dists("data/hja_dendritic-dists.csv")
design$upstreamdist <- as.matrix(den.dists)[1,]

# Read in Distances
# Geo distance Matrix
xy <- cbind(jitter(env$longitude, amount = .0001),
            jitter(env$latitude, amount = .0001))
geo.dists <- SoDA::geoXY(xy[,1], xy[,2])
dist.mat <- fossil::earth.dist(xy) * 1000
```

Next, we will see if any of the environmental variables need to be transformed. I'll then rescale the environmental variables.
```{r}
# Remove orthogonal vectors and make numbers below detection close to zero
env.subs <- env %>% select(habitat, elevation, 
                           temperature, conductivity, 
                           ph, TN, TP, DOC) %>% 
  mutate(TN = if_else(TN < 0, 0.001, TN),
         TP = if_else(TP < 0, 0.001, TP))

#hist(log(env.subs$TP), breaks = 30)
#hist(log(env.subs$TN), breaks = 30)

env.subs <- env.subs %>% mutate(TN = log(TN), TP = log(TP))

# rescale variables
env.subs <- env.subs %>% mutate_if(is_double, scale_vec)
```

Now, I'll perform some transformations on the abundance data. I'll work with the Hellinger-transformed data for the rest of the analysis.
```{r}
# Rarefy communities
set.seed(47405)
OTUs <- rrarefy(OTUs, min(rowSums(OTUs)))
OTUs <- OTUs[,-which(colSums(OTUs) == 0)]

# Transformations and Standardizations
OTUsREL <- decostand(OTUs, method = "total")
OTUs.PA <- decostand(OTUs, method = "pa")
OTUsREL.log <- decostand(OTUs, method = "log")
OTUsREL.hel <- decostand(OTUs, method = "hellinger")
```
I removed the sites with low coverage, and I removed the OTUs with low abundance across the whole dataset.

Here, we will read in the phylogenetic tree. I pruned the phylogenetic tree to match only the taxa remaining in the dataset. Then, I rooted the tree using the midpoint method.
```{r, message=FALSE}
# hja.tree <- read.tree("data/hja_streams.tree")
# matched.phylo <- match.phylo.comm(hja.tree, OTUs)
# hja.tree <- matched.phylo$phy
# is.rooted(hja.tree)
# hja.tree.rooted <- midpoint.root(hja.tree)
# is.rooted(hja.tree.rooted)
# saveRDS(object = hja.tree.rooted, file = "temp/hja_tree_rooted.nwk")
hja.tree.rooted <- readRDS(file = "temp/hja_tree_rooted.nwk")
```


# Environmental analysis
Here, I'll just plot the environmental variables from downstream to upstream across the watershed.
```{r, fig.asp=(1/.618)}
env.subs %>% mutate(upstreamdist = design$upstreamdist, watershed = design$watershed) %>% 
  gather(-upstreamdist, -watershed, -habitat, key = variable, value = measurement) %>% 
  ggplot(aes(x = upstreamdist, y = measurement, color = watershed)) + 
  facet_grid(variable ~ watershed, scales = "free_x") + 
  geom_point() +
  geom_smooth() + 
  theme(legend.position = "top") +
  scale_x_continuous(labels = scales::wrap_format(10)) + 
  labs(x = "Upstream distance (m)", 
       y = "Value (z-score)",
       color = "Watershed")
```

# alpha Diversity analysis
```{r}
alpha.tbl <- tibble(
  habitat = factor(design$habitat, levels = c("sediment", "water"),
                   labels = c("Sediment", "Planktonic")),
  upstream = design$upstreamdist,
  order = design$order,
  sample = rownames(design),
  N0 = rowSums(OTUsREL.hel > 0),
  N1 = exp(diversity(OTUsREL.hel, index = "shannon")),
  N2 = diversity(OTUsREL, index = "invsimpson")
)
summary(lm(N1 ~ habitat, data = alpha.tbl))

alpha.fig <- alpha.tbl %>% 
  ggplot(aes(x = habitat, y = N1, fill = habitat, color = habitat)) +
  geom_boxplot(alpha = 0.8) +
  geom_jitter(alpha = 0.2) +
  labs(x = "", y = expression(paste(alpha, "-diversity (species equivalents)"))) +
  scale_fill_manual(values = (my.colors)) +
  scale_color_manual(values = colorspace::darken(my.colors,.4)) +
  guides(fill = FALSE, color = FALSE) +
  scale_y_continuous(limits = c(0, 3500)) 

summary(lm(N1 ~ habitat * order, data = alpha.tbl))
```

## Unique taxa per habitat
```{r}
sediment.only <- OTUs[, which(
  colSums(OTUs[which(design$habitat == "water"),]) == 0)]
water.only <- OTUs[, which(
  colSums(OTUs[which(design$habitat == "sediment"),]) == 0)]
water.sed <- OTUs[, which(
  colSums(OTUs[which(design$habitat == "sediment"),]) > 1 & 
  colSums(OTUs[which(design$habitat == "water"),]) > 1)]
in.water <- OTUs[, which(
  colSums(OTUs[which(design$habitat == "water"),]) > 1)]
in.water.pa <- decostand(in.water, method = "pa")
in.sediment <- OTUs[, which(
  colSums(OTUs[which(design$habitat == "sediment"),]) > 1)]
in.sediment.pa <- decostand(in.sediment, method = "pa")


# what proportion of taxa are unique at each site
water.richness <- rowSums(in.water.pa[which(design$habitat == "water"),])
# in water samples, but not in any sediments
water.unique <- rowSums(in.water.pa[ 
  which(design$habitat == "water"), # count the number of taxa in each water samp 
  which(colSums(in.water.pa[which(design$habitat == "sediment"),]) == 0)]) # whose abund in sediments is zero
water.unique.frac <- water.unique / water.richness
se(water.unique.frac)

sed.richness <- rowSums(in.sediment.pa[which(design$habitat == "sediment"),])
# in sediments, but not in any water samples
sed.unique <- rowSums(in.sediment.pa[
  which(design$habitat == "sediment"), # count number of taxa in sed samples
  which(colSums(in.sediment.pa[which(design$habitat == "water"),]) == 0)]) #whose abund in water is zero
sed.unique.frac <- sed.unique / sed.richness
se(sed.unique.frac)

unique.fracs <- data.frame("unique_frac" = water.unique.frac, 
                           "habitat" = "Planktonic") %>% 
  rbind.data.frame(., data.frame(
    "unique_frac" = sed.unique.frac, 
    "habitat" = "Sediment"))
unique.fracs <- unique.fracs %>% rownames_to_column(var = "sample")
unique.fracs$habitat <- factor(unique.fracs$habitat, 
                               levels = c("Sediment", "Planktonic"), ordered = F)
unique.fracs <- left_join(unique.fracs, rownames_to_column(design[,c("order","upstreamdist")], var = "sample"))
summary(lm(unique_frac ~ habitat, data = unique.fracs))
summary(lm(unique_frac ~ habitat, data = unique.fracs)) %>% 
  capture.output(file = "tables/unique_compare.txt")

```

```{r}
unique.fig <- unique.fracs %>% 
  ggplot(aes(x = habitat, y = unique_frac, color = habitat, fill = habitat)) + 
  geom_boxplot(alpha = 0.8) +
  geom_jitter(alpha = 0.2) +
  labs(x = "", y = "Proportion habitat-specific taxa") +
  scale_fill_manual(values = (my.colors)) +
  scale_color_manual(values = colorspace::darken(my.colors,.4)) +
  guides(fill = FALSE, color = FALSE) +
  scale_y_continuous(limits = c(0, .3)) 

summary(lm(log10(unique_frac) ~ order*habitat, unique.fracs))
```

```{r}
alpha.to.plot <- unique.fracs %>% left_join(alpha.tbl) %>% select(-N0, -N2) %>% 
  gather(unique_frac, N1, key = "metric", value = "value")
alpha.to.plot$metric <- factor(alpha.to.plot$metric)
levels(alpha.to.plot$metric) <- c(
  expression(paste(alpha, "-diversity")),
  expression(paste("Proportion habitat-specific taxa"))
)
alpha.plot <- alpha.to.plot %>% 
  ggplot(aes(x = habitat, y = value, fill = habitat, color = habitat)) + 
  facet_wrap(~ metric, scales = "free_y", strip.position = "left", labeller = label_parsed, ncol = 2) +
  geom_boxplot(width = .5, alpha = 0.8) +
  geom_jitter(alpha = 0.1) +
  scale_fill_manual(values = (my.colors)) +
  scale_color_manual(values = colorspace::darken(my.colors,0.4)) +
  theme(strip.placement = "outside", legend.position = "none") +
  guides(fill = FALSE) +
  labs(y = "", x = "") 


plot_grid(alpha.fig, unique.fig, ncol = 2, align = "hv", labels = c("A", "B"))
```


# Beta diversity: 
### Ordination
```{r, fig.show='hide'}
hja.pcoa <- run.pcoa(comm = OTUsREL.hel, dist.metric = "euclidean", plot = T)
pcoa.ellipse <- ordiellipse(hja.pcoa$pcoa, str_to_title(design$habitat), display = "sites",
            kind = "se", conf = 0.95, label = T)
pcoa.plot <- cbind.data.frame(vegan::scores(hja.pcoa$pcoa), group = str_to_title(design$habitat))
df_ell <- calc.ellipse(ord = pcoa.plot, ellipse = pcoa.ellipse)


# Run a PERMANOVA
hja.permanova <- adonis(vegdist(OTUsREL, method = "bray") ~ design$habitat + design$order + design$watershed, permutations = 999)
hja.permanova$aov.tab %>% pander::pander()
capture.output(hja.permanova$aov.tab, file = "./tables/hja_permanova.txt")
```

Now, we'll run an RDA. 
```{r}
#hja.rda <- rda(OTUsREL.hel ~ ., env.subs)
#R2.all.vars <- RsquareAdj(hja.rda)$adj.r.squared
#anova(hja.rda, permutations = how(nperm = 999))
#anova(hja.rda, by = "axis", permutations = how(nperm = 999))

hja.dbrda <- dbrda(vegdist(OTUsREL) ~ ., env.subs)

# overall model and 1st two dbRDA axes significant
#anova(hja.dbrda, permutations = how(nperm = 999))
#anova(hja.dbrda, by = "axis", permutations = how(nperm = 999))
```
We see that the RDA is globally significant and the first two canonical axes are also significant. Because the full model was significant, I'll run a model selection procedure.

```{r}
# forward.sel(OTUsREL.hel, 
#             model.matrix(~., env.subs)[,-1],
#             adjR2thresh = R2.all.vars,
#             nperm = 999)

# habitat, elevation, and conductivity were selected by Blanchet 2008 method

# reduced model selects habitat, elevation and conductivity. but I'll add back other key env vars that are typically considered as well. 
hja.rda.reduced <- dbrda(vegdist(OTUsREL) ~ habitat + elevation + conductivity + TP + TN + ph + DOC, env.subs)
```
The model selection procedure left us with the model using only habitat, elevation, and conductivity as predictor variables. Now, we have 3 significant RDA axes with the more parsimonious model. 

# Make figure 2
```{r}
rda.plot <- cbind.data.frame(vegan::scores(hja.rda.reduced)$sites, group = str_to_title(design$habitat), watershed = design$watershed)
rda.var1 <- round(eigenvals(hja.rda.reduced)[1] / sum(eigenvals(hja.rda.reduced)) * 100, 1)
rda.var2 <- round(eigenvals(hja.rda.reduced)[2] / sum(eigenvals(hja.rda.reduced)) * 100, 1)
rda.vecs <- as.data.frame(hja.rda.reduced$CCA$biplot)
rda.vecs$predictor <- c("Planktonic\n habitat", "Elevation", "Cond.",
                        "TP", "TN", "pH", "DOC")
rda.vecs$origin <- 0
scale.arrows = 1
hja.rda.fig <-  ggplot(data = rda.plot, aes(x = dbRDA1, y = dbRDA2)) +
  geom_hline(aes(yintercept = 0), linetype = "dashed", alpha = 0.25, size = 0.25) +
  geom_vline(aes(xintercept = 0), linetype = "dashed", alpha = 0.25, size = 0.25) +
  geom_point(aes(fill = group, shape = watershed), size = 2, alpha = .8) +
  # geom_point(data = subset(rda.plot, group == "Sediment"), color = "black", size = 2) +
  # geom_point(data = subset(rda.plot, group == "Water"), color = "black", size = 2) +
  # geom_path(data = df_ell, 
  #           aes(x = Dim1, y = Dim2, color = group),
  #           size = .8, alpha = 1, linetype = 2) +
  stat_ellipse(data = rda.plot, alpha = 0.7, aes(color = group, linetype = watershed)) +
  labs(x = paste0("dbRDA1 (", rda.var1, "%)"),
    y = paste0("dbRDA2 (", rda.var2, "%)"),
    color = "Habitat", shape = "Watershed") +
  scale_color_manual(values = my.colors) + 
  scale_fill_manual(values = my.colors) +
  scale_shape_manual(values = c(21, 24)) +
  coord_fixed() +
  theme(legend.position = "none") + 
  geom_segment(data = rda.vecs, size = .3,
               aes(x = origin, y = origin,
                   xend = scale.arrows*dbRDA1,
                   yend = scale.arrows*dbRDA2),
               alpha = .7, color = "black",
               arrow = arrow(angle = 20,
                             length = unit(.1, "inches"),
                             type = "open")) +
  geom_text_repel(data = rda.vecs, size = 1.8,
                   aes(x = (scale.arrows + .1)*dbRDA1, 
                       y = (scale.arrows + .1)*dbRDA2, label = predictor),
                   color = "black",
                   segment.alpha = 0,
                  point.padding = 0, 
                  nudge_y = c(-.15, #water
                              0, # elev
                              .035, # conduct
                              0, # TP
                              0, #TN
                              -.05, #ph 
                              .05)) + # DOC
  annotate("text", x = .8, y = 2.2, label = "Benthic", size = 2.5, color = darken(my.colors[1])) +
  annotate("text", x = -.5, y = 2.2, label = "Planktonic", size = 2.5, color = darken(my.colors[2])) +
  ggsave("figures/Fig2.png", width = 8.4, height = 8.4, units = "cm", dpi = 500) +
  ggsave("figures/Fig2.pdf", width = 8.4, height = 8.4, units = "cm")
```

# Subset only flow-connected sites
```{r}
flow.connected <- read.csv("data/flow-connected-matrix.csv", row.names = 1)
den.dists.mat <- as.matrix(den.dists)
flow.connected.dists <- as.dist(
  flow.connected[rownames(den.dists.mat), colnames(den.dists.mat)] * den.dists.mat)
```

# Null model analysis
```{r}
# Calculate abundance-weighted Raup-Crick dissimilarities
regional.abunds <- t(as.matrix(colSums(OTUs)))
regional.relabunds <- decostand(regional.abunds, method = "total")
occupancy.probs <- t(as.matrix(colSums(decostand(OTUs, method = "pa")) / nrow(OTUs)))
site.abunds <- rowSums(OTUs)
site.rich <- specnumber(OTUs)
a <- regional.relabunds * occupancy.probs

# Create a null community based on Stegen et al. 2015
set.seed(47405)
#rc.nulls <- nullcom.rcabund(OTUs = OTUs, stand = "total", distance = "bray")
#saveRDS(rc.nulls, "temp/rc_nullmodels_bray.rda")
rc.nulls <- readRDS("temp/rc_nullmodels_bray.rda")

obs.bray <- as.matrix(vegdist(OTUsREL, method = "bray"))
site.compares <- expand.grid(site1 = 1:nrow(obs.bray), site2 = 1:nrow(obs.bray))
site.compares <- site.compares[-which(site.compares[,1] == site.compares[,2]),]
RC.bray <- matrix(NA, nrow = nrow(obs.bray), ncol = nrow(obs.bray))

for(row.i in 1:nrow(site.compares)){
  site1 <- site.compares[row.i,1]
  site2 <- site.compares[row.i,2]
  pairwise.null <- rc.nulls[site1,site2,]
  pairwise.bray <- obs.bray[site1,site2]
  num.greater <- sum(pairwise.null > pairwise.bray)
  num.ties <- sum(pairwise.null == pairwise.bray)
  val <- -1*((((1 * num.greater) + (0.5 * num.ties))/999 - 0.5) * 2)
  RC.bray[site1, site2] <- val
}
rownames(RC.bray) <- rownames(design)
colnames(RC.bray) <- rownames(design)
RC.bray.dist <- as.dist(RC.bray)
range(RC.bray.dist)

flow_connected_dists_df <- simba::liste(flow.connected.dists, entry = "den_dists")
dists.df <- simba::liste(RC.bray.dist, entry = "RC_bray") %>% add_column(den_dists = flow_connected_dists_df$den_dists)
dists.df$habitat <- NA
dists.df[str_detect(dists.df$NBX, "_W") & str_detect(dists.df$NBY, "_W"),]$habitat <- str_wrap("Planktonic")
dists.df[str_detect(dists.df$NBX, "_W") & str_detect(dists.df$NBY, "_S"),]$habitat <- str_wrap("Planktonic-Benthic")
dists.df[str_detect(dists.df$NBX, "_S") & str_detect(dists.df$NBY, "_W"),]$habitat <- str_wrap("Planktonic-Benthic")
dists.df[str_detect(dists.df$NBX, "_S") & str_detect(dists.df$NBY, "_S"),]$habitat <- str_wrap("Benthic")

dists.df$habitat <- factor(dists.df$habitat, levels = c("Planktonic", "Benthic", "Planktonic-Benthic"))
```

# Match phylo
```{r, results='hide'}
matched.phylo <- match.phylo.comm(phy = hja.tree.rooted, comm = OTUs[,which(colSums(OTUs) > 10)])
hja.comm <- matched.phylo$comm
hja.phy <- matched.phylo$phy
```

# Make figure 4
```{r}
#mntd.hja <- comdistnt.par(hja.comm, cophenetic(hja.phy), abundance.weighted = T, cores = 32)

#hja.mntd.ses <- ses.comdistnt2(
#  samp = hja.comm, 
#  dis = cophenetic(hja.phy), 
#  method = "quasiswap", 
#  fixedmar = "both", 
#  shuffle = "both", 
#  strata = NULL, 
#  mtype = "count", 
#  burnin = 0, 
#  thin = 1, 
#  abundance.weighted = TRUE, 
#  exclude.conspecifics = FALSE, 
#  runs = 999, 
#  cores = 32)
# saveRDS(mntd.hja, file = "data/mntds.rda")

# # Create null comms
# mntd.null <- array(NA, c(50, 50, 999))
# for(i in 1:999){
#   if(i == 1) pb <- progress_bar$new(total = 999, force = T)
#   pb$update(ratio = i/999)
#   #print(paste("creating null community ", i, " of 999"))
#   temp.mntd <- comdistnt(hja.comm,
#                          cophenetic(tipShuffle(hja.phy)),
#                          abundance.weighted=T)
#   mntd.null[,,i] <- as.matrix(temp.mntd)
#   if(i %% 50 == 0 | i == 999) saveRDS(mntd.null, file = "data/mntds-null-dist.rda")
# }

# read null dists 
#mntd.hja <- readRDS(file = "data/mntds.rda")
#mntds.null <- readRDS(file = "data/mntds-null-dist.rda")

# obs.mntds <- as.matrix(mntd.hja)
#site.compares <- expand.grid(site1 = 1:ncol(obs.mntds), site2 = 1:ncol(obs.mntds))
#bNTI <- matrix(NA, nrow = nrow(obs.mntds), ncol = ncol(obs.mntds))
#for(row.i in 1:nrow(site.compares)){
#  site1 <- site.compares[row.i,1]
#  site2 <- site.compares[row.i,2]
#  pairwise.null <- mntds.null[site1,site2,]
#  pairwise.mntd <- obs.mntds[site1,site2]
#  null.mean <- mean(pairwise.null, na.rm = TRUE)
#  null.sd <- sd(pairwise.null, na.rm = TRUE)
#  val <- (pairwise.mntd - null.mean) / null.sd
#  bNTI[site1, site2] <- val
#}
#colnames(bNTI) <- rownames(hja.comm)
#rownames(bNTI) <- rownames(hja.comm)



hja.mntd.ses <- readRDS("temp/hja.mntd.ses.rda")
bNTI.dist <- as.dist(hja.mntd.ses$comdistnt.obs.z)
sum(bNTI.dist < 2 & bNTI.dist > -2) / length(bNTI.dist) # undom
sum(bNTI.dist > 2) / length(bNTI.dist) # variable selection
sum(bNTI.dist < -2) / length(bNTI.dist) # homogeneous selection


hja.bnti.dist.ls <- simba::liste(bNTI.dist, entry = "bNTI")
hja.rcbray.dist.ls <- simba::liste(RC.bray.dist, entry = "RCbray")
hja.euclid.dist.ls <- simba::liste(dist(SoDA::geoXY(latitude = xy[,2], longitude = xy[,1])), entry = "euclidean")
hja.assembly <- full_join(dists.df, hja.bnti.dist.ls)
hja.assembly <- cbind(hja.assembly, env = simba::liste(dist(env.subs[,-1]))[,3], euclid = hja.euclid.dist.ls$euclidean)


hja.assembly.plot <- hja.assembly %>% filter(den_dists > 0) %>% 
  mutate(process = ifelse(bNTI < -2, "Homogeneous selection", 
                          ifelse(bNTI > 2, "Variable selection",
                                 ifelse(RC_bray > 0.95, "Dispersal limitation",
                                        ifelse(RC_bray < -0.95, "Mass effects", "Undominated"))))) %>% 
  mutate(signif = ifelse(abs(RC_bray) < 0.95 & abs(bNTI) < 2, FALSE, TRUE)) %>% 
  gather(RC_bray, bNTI, key = "metric", value = "value") 
hja.assembly.plot$metric <- factor(hja.assembly.plot$metric)
levels(hja.assembly.plot$metric) <- c(
  expression(paste(beta,"NTI")),
  expression(beta["RC, Bray-Curtis"])
)

hja.assembly.rounded <- hja.assembly.plot %>% 
  mutate(den_dists_log10 = log10(den_dists)) %>% 
  mutate(den_dists_log10_rounded = round(den_dists_log10,0)) %>% 
  mutate(den_dists_rounded = 10^den_dists_log10_rounded)


hja.assembly.grouped <- hja.assembly.rounded %>% 
  group_by(habitat, den_dists_rounded) %>% 
  count(process) %>% 
  mutate(proportion = n/sum(n)) 
hja.assembly.grouped <- hja.assembly.grouped %>% 
  full_join(unique(expand.grid(hja.assembly.grouped[,1:3])), fill = 0) %>% 
  mutate(proportion = ifelse(is.na(proportion),0, proportion))

write_csv(hja.assembly.grouped, "tables/assembly_counts_by_scale.csv")

hja.assembly.grouped %>% 
  arrange(process, den_dists_rounded) %>% 
  mutate(process = factor(process, levels = c("Undominated", 
                                              "Variable selection",
                                              "Homogeneous selection",
                                              "Mass effects",
                                              "Dispersal limitation"))) %>% 
  ggplot(aes(x = den_dists_rounded, y = proportion, fill = process)) +
  geom_area(alpha = 1, position = "stack", color = NA) +
  scale_x_log10() +
  facet_wrap(~ habitat, ncol = 1) + 
  scale_fill_manual(values = c("#02440c","#789400","#e0d052", "#edf0db", "#40becd")) + 
  labs(x = "Dendritic distance (m)", y = "Proportion") +
  theme(legend.position = "bottom", 
        legend.title = element_blank(),
        legend.text = element_text(size = 9)) +
  guides(fill = guide_legend(nrow = 2)) +
  ggsave("figures/Fig4.png", height = 8, width = 4, dpi = 600) + 
  ggsave("figures/Fig4.pdf", height = 8, width = 4)

hja.assembly.grouped %>% 
  ungroup() %>% 
  group_by(den_dists_rounded) %>% 
  count()

hja.assembly.grouped %>% 
    count()

hja.assembly.plot %>% 
  group_by(habitat) %>% 
  count(process)
hja.assembly.plot %>% count(process)
```

# Make figure 5
```{r}
headwaters.dist <- as.dist(as.matrix(bNTI.dist)[which(design$order == 1), which(design$order == 1)])
mainstem.dist <- as.dist(as.matrix(bNTI.dist)[which(design$order != 1), which(design$order != 1)])

headwaters.dist.sed.tax <- as.dist(as.matrix(RC.bray.dist)[
  which(design$order == 1 & design$habitat != "water"), 
  which(design$order == 1 & design$habitat != "water")])
mainstem.dist.sed.tax <- as.dist(as.matrix(RC.bray.dist)[
  which(design$order != 1 & design$habitat != "water"), 
  which(design$order != 1 & design$habitat != "water")])
headwaters.dist.water.tax <- as.dist(as.matrix(RC.bray.dist)[
  which(design$order == 1 & design$habitat == "water"), 
  which(design$order == 1 & design$habitat == "water")])
mainstem.dist.water.tax <- as.dist(as.matrix(RC.bray.dist)[
  which(design$order != 1 & design$habitat == "water"), 
  which(design$order != 1 & design$habitat == "water")])

headwaters.dist.sed.phy <- as.dist(as.matrix(bNTI.dist)[
  which(design$order == 1 & design$habitat != "water"), 
  which(design$order == 1 & design$habitat != "water")])
mainstem.dist.sed.phy <- as.dist(as.matrix(bNTI.dist)[
  which(design$order != 1 & design$habitat != "water"), 
  which(design$order != 1 & design$habitat != "water")])
headwaters.dist.water.phy <- as.dist(as.matrix(bNTI.dist)[
  which(design$order == 1 & design$habitat == "water"), 
  which(design$order == 1 & design$habitat == "water")])
mainstem.dist.water.phy <- as.dist(as.matrix(bNTI.dist)[
  which(design$order != 1 & design$habitat == "water"), 
  which(design$order != 1 & design$habitat == "water")])

# now cross-habitat comparisons, not square matrix
headwaters.dist.sed.water.phy <- as.matrix(bNTI.dist)[
  which(design$order == 1 & design$habitat != "water"), 
  which(design$order == 1 & design$habitat == "water")]
downstream.dist.sed.water.phy <- as.matrix(bNTI.dist)[
  which(design$order != 1 & design$habitat != "water"), 
  which(design$order != 1 & design$habitat == "water")]

headwaters.dist.sed.water.tax <- as.matrix(RC.bray.dist)[
  which(design$order == 1 & design$habitat != "water"), 
  which(design$order == 1 & design$habitat == "water")]
downstream.dist.sed.water.tax <- as.matrix(RC.bray.dist)[
  which(design$order != 1 & design$habitat != "water"), 
  which(design$order != 1 & design$habitat == "water")]

# construct dfs for headwater downstream comparison
hwsp <- simba::liste(headwaters.dist.sed.phy, entry = "bNTI")
hwwp <- simba::liste(headwaters.dist.water.phy, entry = "bNTI")
dssp <- simba::liste(mainstem.dist.sed.phy, entry = "bNTI")
dswp <- simba::liste(mainstem.dist.water.phy, entry = "bNTI")
hwswp <- simba::liste(headwaters.dist.sed.water.phy, entry = "bNTI")
dsswp <- simba::liste(downstream.dist.sed.water.phy, entry = "bNTI")

hwswt <- simba::liste(headwaters.dist.sed.water.tax, entry = "RC_hel")
dsswt <- simba::liste(downstream.dist.sed.water.tax, entry = "RC_hel")
hwst <- simba::liste(headwaters.dist.sed.tax, entry = "RC_hel")
hwwt <- simba::liste(headwaters.dist.water.tax, entry = "RC_hel")
dsst <- simba::liste(mainstem.dist.sed.tax, entry = "RC_hel")
dswt <- simba::liste(mainstem.dist.water.tax, entry = "RC_hel")

hwsp$position <- "Headwater"
hwwp$position <- "Headwater"
hwswp$position <- "Headwater"
hwst$position <- "Headwater"
hwwt$position <- "Headwater"
hwswt$position <- "Headwater"

dssp$position <- "Downstream"
dswp$position <- "Downstream"
dsswp$position <- "Downstream"
dsst$position <- "Downstream"
dswt$position <- "Downstream"
dsswt$position <- "Downstream"

hwsp$habitat <- "Benthic"
hwwp$habitat <- "Planktonic"
hwswp$habitat <- "Planktonic-Benthic"
hwst$habitat <- "Benthic"
hwwt$habitat <- "Planktonic"
hwswt$habitat <- "Planktonic-Benthic"

dssp$habitat <- "Benthic"
dswp$habitat <- "Planktonic"
dsswp$habitat <- "Planktonic-Benthic"
dsst$habitat <- "Benthic"
dswt$habitat <- "Planktonic"
dsswt$habitat <- "Planktonic-Benthic"

assembly.by.position <- bind_rows(
  full_join(hwsp, hwst),
  full_join(hwwp, hwwt),
  full_join(hwswp, hwswt),
  full_join(dssp, dsst),
  full_join(dswp, dswt),
  full_join(dsswp, dsswt)
)

assembly.by.position.process <- assembly.by.position %>% 
  mutate(process = ifelse(bNTI < -2, "Homogeneous selection", 
                          ifelse(bNTI > 2, "Variable selection",
                                 ifelse(RC_hel > 0.95, "Dispersal limitation",
                                        ifelse(RC_hel < -0.95, "Mass effects", "Undominated"))))) %>% 
  mutate(signif = ifelse(abs(RC_hel) < 0.95 & abs(bNTI) < 2, FALSE, TRUE)) 


assembly.position.grouped <- assembly.by.position.process %>% 
  group_by(habitat, position) %>% 
  count(process) %>% 
  mutate(proportion = n/sum(n)) 

assembly.position.grouped <- assembly.position.grouped %>% 
  full_join(unique(expand.grid(assembly.position.grouped[,1:3])), fill = 0) %>% 
  mutate(proportion = ifelse(is.na(proportion),0, proportion))

assembly.position.grouped$habitat <- factor(assembly.position.grouped$habitat, levels = c("Planktonic", "Benthic", "Planktonic-Benthic"))

assembly.position.grouped %>% 
  mutate(position = factor(position, levels = c("Headwater", "Downstream"))) %>% 
  mutate(process = factor(process, levels = c("Undominated",
                                              "Variable selection",
                                              "Homogeneous selection",
                                              "Mass effects",
                                              "Dispersal limitation"))) %>%
  ggplot(aes(x = position, y = proportion, fill = process)) +
  geom_bar(alpha = 1, position = "stack", stat = "identity") +
  facet_wrap(~ habitat, ncol = 1) +
  scale_fill_manual(values = c("#02440c","#789400","#e0d052", "#edf0db", "#40becd")) + 
  labs(x = "", y = "Proportion") +
  theme(legend.position = "bottom", 
        legend.title = element_blank(),
        legend.text = element_text(size = 9)) +
  guides(fill = guide_legend(nrow = 2)) +
  ggsave("figures/Fig5.png", height = 8, width = 4, dpi = 600) + 
  ggsave("figures/Fig5.pdf", height = 8, width = 4)

assembly.position.grouped %>% count() 
assembly.by.position.process %>% 
  group_by(position, habitat) %>% 
  count(process)
```

# Make figure 3
```{r}
hja.assembly$hel <- vegdist(OTUsREL.hel, method = "euc")
hja.assembly$bray <- vegdist(OTUsREL, method = "bray")

hja.assembly %>% left_join(simba::liste(den.dists, entry = "unconnected_dists")) %>% 
  ggplot(aes(x = unconnected_dists +1, y = bray, color = habitat, fill = habitat, shape = habitat)) +
  geom_point(alpha = 0.25) +
  geom_smooth(method = "lm") +
  scale_x_log10(lim = c(1, 50000)) +
  scale_color_manual(values=colorspace::darken(c(my.colors, "#688c45"),0.4)) +
  scale_fill_manual(values=c(my.colors, "#688c45", my.colors,	"#688c45")) +
  labs(x = "Dendritic distance (m)",
       y = "Community dissimilarity") +
  theme(legend.position = "none") +
  annotate("text", x = 45000, y = 0.75, label = "Benthic-\nPlanktonic", size = 1.5, color = darken("#688c45")) +
  annotate("text", x = 45000, y = 0.67, label = "Planktonic", size = 1.5, color = darken("#3a4f6a")) +
  annotate("text", x = 45000, y = 0.62, label = "Benthic", size = 1.5, color = darken("#d5ba94",.4)) +
  ggsave("figures/Fig3.pdf", width = 8, height = 8*3/4, units = "cm") +
  ggsave("figures/Fig3.png", width = 8, height = 8*3/4, units = "cm", dpi = 600)

ddr_formod <- hja.assembly %>% left_join(simba::liste(den.dists, entry = "unconnected_dists")) %>% mutate(unconnected_dists = unconnected_dists + 1)

ddr_envdist_mod <- lm(bray ~ log10(unconnected_dists+1) * habitat + env, data = ddr_formod)
AIC(ddr_envdist_mod)
plot(ddr_envdist_mod)
summary(ddr_envdist_mod)
```

```{r}
HF00402_v12 <- read_csv("data/HF00402_v12.csv")

HF00402_v12 %>% 
  filter(SITECODE == "GSLOOK") %>% 
  filter(lubridate::year(DATE) == "2015") %>% 
  ggplot(aes(x = DATE, y = MEAN_Q)) + 
  geom_line() +
  theme(panel.grid.major = element_line(size = .2, color = "gray90"),
        panel.grid.minor = element_line(size = .1, color = "gray90")) +
  labs(x="", y = "Mean discharge (cfs)", subtitle = "Annual hydrograph for Lookout Creek watershed, H.J. Andrews LTER")
    
```

\newpage
# Appendix: Session Info
```{r}
sessionInfo()
```

