---
title: "Analysis for Stream Microbial Metacommunity"
output: pdf_document
---
### Initial Setup
```{r}
rm(list=ls())
setwd("~/GitHub/HJA-streams/analysis")
opar <- par()
```

### Import Source Files and Packages
```{r}
source("MothurTools.R")

require("vegan")
require("png")
require("raster")
require("sp")
require("rgdal")
require("SoDA")
require("grid")
require("simba")

se <- function(x, ...){sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))}

error.bar <- function(x, y, upper, lower=upper, length=0.1,...){
  if(length(x) != length(y) | length(y) !=length(lower) | length(lower) != length(upper))
    stop("vectors must be same length")
  arrows(x,y+upper, x, y-lower, angle=90, code=3, length=length, ...)
}

```


### Import Shared, Design, and Environment Files
```{r}
# Define Inputs
# Design = general design file for experiment
# shared = OTU table from mothur with sequence similarity clustering
# Taxonomy = Taxonomic information for each OTU
design <- "../data/design.txt"
shared <- "../data/hja_streams.final.shared"
taxon  <- "../data/hja_streams.final.0.03.taxonomy"
env    <- "../data/hja_env.csv"

# Import Design
design <- read.delim(design, header=T, row.names=1)

# Import Shared Files
OTUs <- read.otu(shared = shared, cutoff = "0.03")         # 97% Similarity

# Import Taxonomy
OTU.tax <- read.tax(taxonomy = taxon, format = "rdp")

# Import Env
env.total <- read.csv(env, header=T)

```

### Data Transformations
```{r}
# Remove OTUs with less than two occurances across all sites
OTUs <- OTUs[, which(colSums(OTUs) >= 2)]

# Sequencing an Good's Coverage
# Sequencing Coverage
coverage <- rowSums(OTUs)

# Good's Coverage
goods <- function(x = ""){
  1 - (sum(x == 1) / rowSums(x))
}
goods.c <- goods(OTUs)

# Remove Low Coverage Samples
lows <- which(coverage < 7000)
lows
OTUs <- OTUs[-which(coverage < 7000), ]
design <- design[-which(coverage < 7000), ]
env <- env.total[-which(coverage < 7000), ]

# Remove orthogonal vectors
env <- env[c(1:11, 13, 16, 18, 19)]
env.mat <- as.matrix(env[10:15])
env.mat[51,5] <- 150
env.mat[52,5] <- 150
for(i in 1:nrow(env.mat)){
  if(env.mat[i, 5] < 0){
    env.mat[i, 5] <- 0.0001
  }
  if(env.mat[i, 6] < 0){
    env.mat[i, 6] <- 0.0001
  }
}
env.mat <- scale(env.mat)

# Distance Matrix
xy <- cbind(env$longitude, env$latitude)
geo.dists <- geoXY(env$latitude, env$longitude)
xy <- project(xy, "+proj=utm +zone=10 +ellps=WGS84")
dist.mat <- as.matrix(dist(xy, method = "euclidean"))

# Connectivity Matrix 
adj.mat <- as.matrix(read.csv("../data/undirected-matrix.csv", header=T))
row.names(adj.mat) <- adj.mat[,1]
adj.mat <- adj.mat[,-1]
adj.mat <- (adj.mat == 1) * 1
adj.mat <- adj.mat[-which(coverage < 7000), -which(coverage < 7000)]

# Make Relative Abundence Matrices
OTUsREL <- decostand(OTUs, method = "total")

# Transform Relative Abundances
OTUsREL.log <- decostand(OTUs, method = "log")

OTUs.PA <- decostand(OTUs, method = "pa")

OTUsREL.hel <- decostand(OTUs, method = "hellinger")

```

### Calculate Alpha Diversity
```{r}
# Observed Richness
S.obs <- rowSums((OTUs > 0) * 1)

# Simpson's Evenness
SimpE <- function(x = ""){
  x <- as.data.frame(x)
  D <- diversity(x, "inv")
  S <- sum((x > 0) * 1) 
  E <- (D)/S 
  return(E)
}
simpsE <- round(apply(OTUsREL.hel, 1, SimpE), 3)

shan <- diversity(OTUsREL.hel, index = "shannon")
simpsD <- diversity(OTUsREL.hel, index = "invsimpson")


# Rarefaction
hja.S <- rowSums((OTUs > 0) * 1)
min.N <- min(rowSums(OTUs))
S.rarefy <- rarefy(x = OTUs, sample = (min.N/2), se = TRUE)
#S.rarefy2 <- rarefy(x = OTUs, sample = min.N/2, se = TRUE)
#rarecurve(x = OTUs, step = 20, col = "blue", cex = 0.6, las = 1)
rare.d <- t(S.rarefy)
#rare.d.2 <- t(S.rarefy2)
colnames(rare.d) <- c("S.rare", "se.rare")
#colnames(rare.d.2) <- c("S.rare.sub", "se.rare.sub")


alpha.div <- cbind(design, S.obs, simpsE, shan, simpsD, rare.d)
```


#### Alpha Diversity Across Habitats
```{r}
# Seperate data based on water and sediment samples
water <- alpha.div[alpha.div$habitat == "water",]
sed <- alpha.div[alpha.div$habitat == "sediment", ]

lc   <- alpha.div[alpha.div$watershed == "LC",]
ws01 <- alpha.div[alpha.div$watershed == "WS01",]

# Differences in diversity (richness and evenness) between habitats
t.test(S.rare ~ habitat, data = alpha.div)
mean(c(358.0118, 120.1305)) / mean(alpha.div$S.rare[alpha.div$habitat == "sediment"])


t.test(shan ~ habitat, data = alpha.div)
t.test(simpsE ~ habitat, data = alpha.div)

plot(simpsE ~ order, data = alpha.div[which(alpha.div$habitat == "water"),])
a.1 <- alpha.div[which(alpha.div$order == 1),]
a.2 <- alpha.div[which(alpha.div$order != 1),]

t.test(a.1$S.rare, a.2$S.rare)

# Rare tail:
cor.test(alpha.div$S.obs, alpha.div$simpsE)
#plot(alpha.div$S.obs, alpha.div$simpsE)
```

### Figure 1: Rarefied diversity in sediment and surface-water communities
```{r}
png(filename = "../figures/Figure1.png",
    width = 1200, height = 1200, res = 96*2)
par(mar = c(5, 5, 3, 2) + 0.3)
boxplot(S.rare ~ habitat, data = alpha.div, at = c(3,1),
     names = c("Sediment", "Water"), col = c("grey", "white"),
     xlim = c(0,4),
     yaxt = "n", ylab = "", xlab = "", cex.axis = 1.2)
axis(side = 2, labels = T, lwd.ticks = 2, cex.axis = 1.2, las = 1)
axis(side = 1, labels = F, at = c(1,3), lwd.ticks = 2, cex.axis = 1.2, las = 1)
box(lwd = 2)
text(3, 1700, "*", cex = 2)
mtext("Habitat Type", side = 1, line = 3, cex = 1.5)
mtext("Species Richness", side = 2, line = 4, cex = 1.5)
dev.off()
graphics.off()

```

```{r fig.width=4, fig.height=4,echo=FALSE,fig.cap="Higher richness in surface water than in sediments."}
img <- readPNG("../figures/Figure1.png")
grid.raster(img)
```

#### Sed-water comparisons within sites
```{r}
paired.div <- alpha.div
paired.OTUsREL.log <- OTUsREL.log
remove.sites <- 0
for(site in unique(paired.div$site)){
  if(sum(paired.div$site == site) < 2){
    paired.div <- paired.div[-which(paired.div$site == site),]
    remove.sites <- c(remove.sites, which(design$site == site))
  }
}
paired.OTUsREL.log <- paired.OTUsREL.log[-remove.sites,]

paired.div.difs <- (paired.div$S.rare[which(paired.div$habitat == "water")] -
        paired.div$S.rare[which(paired.div$habitat == "sediment")])
mean(paired.div.difs)
se(paired.div.difs)
t.test(paired.div$S.rare[which(paired.div$habitat == "water")],
        paired.div$S.rare[which(paired.div$habitat == "sediment")], paired = TRUE)
```

#### Figure 2: Water vs. Sed. diversity
```{r}
png(filename = "../figures/Figure2.png",
    width = 1200, height = 1200, res = 96*2)
par(c(5,5,3,2) + 0.3)
barplot(height = sort(paired.div.difs, decreasing = T), names.arg = unique(paired.div$site),
        horiz = TRUE, yaxt = "n", xaxt = "n", xlim = c(-1500,1500),
        col = "black", border = NA)
axis(side = 1, labels = T, lwd = 2, lwd.ticks = 2)
axis(side = 3, labels = F, lwd = 2, tck = 0)
abline(v = 0, lty = 3)
mtext(side = 1, line = 3, cex = 1.5, expression(paste("Difference in ",alpha,"-diversity")))
mtext(side = 3, line = 0.5, at = c(-1500), adj = 0, "Higher in\nsediment")
mtext(side = 3, line = 0.5, at = c(1500), adj = 1, "Higher in\nwater")
#mtext(side = 2, line = 2.5, cex = 1.2, "Site")
dev.off()
graphics.off()


```

```{r fig.width=4, fig.height=4,echo=FALSE,fig.cap="At a site, alpha diversity tends to be higher in water than sediments."}
img <- readPNG("../figures/Figure2.png")
grid.raster(img)
```

### Calculate Bray-Curtis across stream orders
```{r}
order.1.bray.w <- vegdist(OTUsREL.log[which(design$order==1 & design$habitat == "water"),], method = "bray")
order.2.bray.w <- vegdist(OTUsREL.log[which(design$order==2 & design$habitat == "water"),], method = "bray")
order.3.bray.w <- vegdist(OTUsREL.log[which(design$order==3 & design$habitat == "water"),], method = "bray")
order.4.bray.w <- vegdist(OTUsREL.log[which(design$order==4 & design$habitat == "water"),], method = "bray")
order.5.bray.w <- vegdist(OTUsREL.log[which(design$order==5 & design$habitat == "water"),], method = "bray")

order.1.bray.s <- vegdist(OTUsREL.log[which(design$order==1 & design$habitat == "sediment"),], method = "bray")
order.2.bray.s <- vegdist(OTUsREL.log[which(design$order==2 & design$habitat == "sediment"),], method = "bray")
order.3.bray.s <- vegdist(OTUsREL.log[which(design$order==3 & design$habitat == "sediment"),], method = "bray")
order.4.bray.s <- vegdist(OTUsREL.log[which(design$order==4 & design$habitat == "sediment"),], method = "bray")
order.5.bray.s <- vegdist(OTUsREL.log[which(design$order==5 & design$habitat == "sediment"),], method = "bray")

order.1.dists.w <- as.vector(unlist(list(order.1.bray.w)))
order.2.dists.w <- as.vector(unlist(list(order.2.bray.w)))
order.3.dists.w <- as.vector(unlist(list(order.3.bray.w)))
order.4.dists.w <- as.vector(unlist(list(order.4.bray.w)))
order.5.dists.w <- as.vector(unlist(list(order.5.bray.w)))

order.1.dists.s <- as.vector(unlist(list(order.1.bray.s)))
order.2.dists.s <- as.vector(unlist(list(order.2.bray.s)))
order.3.dists.s <- as.vector(unlist(list(order.3.bray.s)))
order.4.dists.s <- as.vector(unlist(list(order.4.bray.s)))
order.5.dists.s <- as.vector(unlist(list(order.5.bray.s)))


orders.bray.w <- as.data.frame(cbind(c(
        rep(1, length(order.1.dists.w)),
        rep(2, length(order.2.dists.w)),
        rep(2, length(order.3.dists.w)),
        rep(2, length(order.4.dists.w)),
        rep(2, length(order.5.dists.w))),
      c(order.1.dists.w, order.2.dists.w, order.3.dists.w,
        order.4.dists.w, order.5.dists.w)))
colnames(orders.bray.w) <- c("order", "dist")
model.orders.w <- lm(orders.bray.w$dist ~ orders.bray.w$order)
(modsum.w <- summary(model.orders.w))

orders.bray.s <- as.data.frame(cbind(c(
        rep(1, length(order.1.dists.s)),
        rep(2, length(order.2.dists.s)),
        rep(2, length(order.3.dists.s)),
        rep(2, length(order.4.dists.s)),
        rep(2, length(order.5.dists.s))),
      c(order.1.dists.s, order.2.dists.s, order.3.dists.s,
        order.4.dists.s, order.5.dists.s)))
colnames(orders.bray.s) <- c("order", "dist")
model.orders.s <- lm(orders.bray.s$dist ~ orders.bray.s$order)
(modsum.s <- summary(model.orders.s))

model.orders.w <- aov(orders.bray.w$dist ~ as.factor(orders.bray.w$order))
(modsum.w <- summary(model.orders.w))
TukeyHSD(model.orders.w)

model.orders.s <- aov(orders.bray.s$dist ~ as.factor(orders.bray.s$order))
(modsum.s <- summary(model.orders.s))
TukeyHSD(model.orders.s)
```

# Figure 3: Beta-diversity is higher among headwaters than among higher order streams.
```{r}
png(filename = "../figures/Figure3.png",
    width = 1200, height = 1200, res = 96*2)
par(mar = c(5, 5, 2, 2) + 0.1)
boxplot(dist ~ order, data = orders.bray.w, at = c(1,4), xlim = c(0,6), ylim = c(0.4, 0.9), 
        col = "white", xlab = "", xaxt = "n", yaxt = "n")
boxplot(dist ~ order, data = orders.bray.s, at = c(2,5), add = T, col = "grey",
        xlab = "", xaxt = "n", yaxt = "n")
axis(side = 1, labels = F, at = c(1.5, 4.5), lwd.ticks = 2, cex.axis = 1.2, las = 1)
axis(side = 2, labels = T, lwd.ticks = 2, cex.axis = 1.2, las = 1)
axis(side = 3, labels = F, lwd.ticks = 2, cex.axis = 1.2, las = 1)
axis(side = 4, labels = F, lwd.ticks = 2, cex.axis = 1.2, las = 1)
box(lwd=2)
mtext(side = 2, cex = 1.5, line = 3.5, expression(paste(beta,"-diversity")))
mtext(side = 1, cex = 1.5, line = 3, "Stream Order")
mtext(side = 1, at = c(1.5,4.5), line = 1, c("Headwaters","Higher-order"), cex = 1.2)

legend("topright", c("Water", "Sediment"),
         pt.bg = c("white", "grey"), pch = c(22,22), cex = 1.5, bty = "n")

dev.off()
graphics.off()
```

```{r fig.width=4, fig.height=4,echo=FALSE,fig.cap="Microbial Commuinity Shifts"}
img <- readPNG("../figures/Figure3.png")
grid.raster(img)
```
### Ordination

#### PCoA 
```{r}
# All HJA Catchment
hja.db <- vegdist(OTUsREL, method = "bray", upper = TRUE, diag = TRUE)
hja.pcoa <- cmdscale(hja.db, eig=TRUE, k=3)
var1 <- round(hja.pcoa$eig[1] / sum(hja.pcoa$eig),3) * 100
var2 <- round(hja.pcoa$eig[2] / sum(hja.pcoa$eig),3) * 100
var3 <- round(hja.pcoa$eig[3] / sum(hja.pcoa$eig),3) * 100

# Sediments Only
sediment <- OTUsREL.log[which(design$habitat == "sediment"),]
sediment.db <- vegdist(sediment, method = "bray", diag = T)
sediment.pcoa <- cmdscale(sediment.db, eig=TRUE, k=3)
sed.design <- design[which(design$habitat == "sediment"),]
s.var1 <- round(sediment.pcoa$eig[1] / sum(sediment.pcoa$eig),3) * 100
s.var2 <- round(sediment.pcoa$eig[2] / sum(sediment.pcoa$eig),3) * 100
s.var3 <- round(sediment.pcoa$eig[3] / sum(sediment.pcoa$eig),3) * 100

# Water Only
water <- OTUsREL.log[which(design$habitat == "water"),]
water.db <- vegdist(water, method = "bray", diag = T)
water.pcoa <- cmdscale(water.db, eig=TRUE, k=3)
water.design <- design[which(design$habitat == "water"),]
w.var1 <- round(water.pcoa$eig[1] / sum(water.pcoa$eig),3) * 100
w.var2 <- round(water.pcoa$eig[2] / sum(water.pcoa$eig),3) * 100
w.var3 <- round(water.pcoa$eig[3] / sum(water.pcoa$eig),3) * 100

# Lookout Creek Watershed Only
lc <- OTUsREL[which(design$watershed == "LC"),]
lc.db <- vegdist(lc, method = "bray", diag = T)
lc.pcoa <- cmdscale(lc.db, eig=TRUE, k=3)
lc.design <- design[which(design$watershed == "LC"),]
lc.var1 <- round(lc.pcoa$eig[1] / sum(lc.pcoa$eig),3) * 100
lc.var2 <- round(lc.pcoa$eig[2] / sum(lc.pcoa$eig),3) * 100
lc.var3 <- round(lc.pcoa$eig[3] / sum(lc.pcoa$eig),3) * 100

# Watershed 01 Only
w1 <- OTUsREL[which(design$watershed == "WS01"),]
w1.db <- vegdist(w1, method = "bray", diag = T)
w1.pcoa <- cmdscale(w1.db, eig=TRUE, k=3)
w1.design <- design[which(design$watershed == "WS01"),]
w1.var1 <- round(w1.pcoa$eig[1] / sum(w1.pcoa$eig),3) * 100
w1.var2 <- round(w1.pcoa$eig[2] / sum(w1.pcoa$eig),3) * 100
w1.var3 <- round(w1.pcoa$eig[3] / sum(w1.pcoa$eig),3) * 100
```

#### Figure 4: Sediments & Water Communities PCoA
```{r}
png(filename = "../figures/Figure4.png",
    width = 1200, height = 1200, res = 96*2)

par(mar = c(5, 5, 3, 2) + 0.1)
plot(hja.pcoa$points[ ,1], hja.pcoa$points[ ,2], ylim = c(-0.5, 0.4), xlim = c(-.35, .35),
     xlab = paste("PCoA 1 (", var1, "%)", sep = ""),
     ylab = paste("PCoA 2 (", var2, "%)", sep = ""), 
     pch = 19, cex = 2.0, type = "n", cex.lab = 1.5, cex.axis = 1.2, axes = F)
axis(side = 1, labels = T, at = c(-0.3,-.15,0,.15,.3), lwd.ticks = 2, cex.axis = 1.2, las = 1)
axis(side = 2, labels = T, lwd.ticks = 2, cex.axis = 1.2, las = 1)
axis(side = 3, labels = F, at = c(-0.3,-.15,0,.15,.3), lwd.ticks = 2, cex.axis = 1.2, las = 1)
axis(side = 4, labels = F, lwd.ticks = 2, cex.axis = 1.2, las = 1)
abline(h = 0, v = 0, lty = 3)
box(lwd = 2)
points(hja.pcoa$points[which(design$habitat == "sediment"),1], 
       hja.pcoa$points[which(design$habitat == "sediment"),2],
       pch=21, cex=2, bg="grey")
points(hja.pcoa$points[which(design$habitat == "water"),1], 
       hja.pcoa$points[which(design$habitat == "water"),2],
       pch=24, cex=2, bg="white")
legend("topright", c("Water", "Sediment"),
         pt.bg = c("white", "grey"), pch = c(24,21), cex = 1.5, bty = "n")

dev.off()
graphics.off()


```

```{r fig.width=4, fig.height=4,echo=FALSE,fig.cap="Ordination across the HJ Andrews Catchment"}
img <- readPNG("../figures/Figure4.png")
grid.raster(img)
```

# Important drivers of community structure
```{r}
# hja.ano <- anosim(hja.db, grouping = design$habitat)
# summary(hja.ano)
# plot(hja.ano)

adonis(hja.db ~ design$order, permutations = 999)
adonis(hja.db ~ design$habitat * design$watershed + design$order, permutations = 999)

```

### Identify Unique OTUs across habitats
```{r}
sediment.only <- OTUs[, which(colSums(OTUs[which(design$habitat == "water"),]) == 0)]
water.only <- OTUs[, which(colSums(OTUs[which(design$habitat == "sediment"),]) == 0)]
water.sed <- OTUs[, which(colSums(OTUs[which(design$habitat == "sediment"),]) > 1 & 
                            colSums(OTUs[which(design$habitat == "water"),]) > 1)]
in.water <- OTUs[, which(colSums(OTUs[which(design$habitat == "water"),]) > 1)]
in.water.pa <- (in.water > 0) * 1
in.sediment <- OTUs[, which(colSums(OTUs[which(design$habitat == "sediment"),]) > 1)]
in.sediment.pa <- (in.sediment > 0) * 1

water.frac <- as.vector(rowSums(in.water.pa[which(design$habitat == "water"), which(colSums(in.water.pa[which(design$habitat == "sediment"),]) > 0)]) / rowSums(in.water.pa[which(design$habitat == "water"),]))
se(rowSums(in.water[which(design$habitat == "water"), which(colSums(in.water[which(design$habitat == "sediment"),]) > 1)]) / rowSums(in.water[which(design$habitat == "water"),]))

sed.frac <- as.vector(rowSums(in.sediment.pa[which(design$habitat == "sediment"), which(colSums(in.sediment.pa[which(design$habitat == "water"),]) > 1)]) / rowSums(in.sediment.pa[which(design$habitat == "sediment"),]))
se(rowSums(in.sediment[which(design$habitat == "sediment"), which(colSums(in.sediment[which(design$habitat == "water"),]) > 1)]) / rowSums(in.sediment[which(design$habitat == "sediment"),]))
```

### Figure 5: Proportion unique taxa
```{r}
png(filename = "../figures/Figure5.png",
    width = 1200, height = 1200, res = 96*2)
par(mar = c(5, 5, 2, 2) + 0.1)
boxplot((1-water.frac), (1-sed.frac), names = c("Water", "Sediments"), col = c("white", "grey"),
        at = c(1,3), xlim = c(0,4), yaxt = "n", cex.axis = 1.2)
axis(side = 2, labels = T, lwd.ticks = 2, cex.axis = 1.2, las = 1)
axis(side = 1, labels = F, at = c(1,3), lwd.ticks = 2, cex.axis = 1.2, las = 1)
box(lwd=2)
mtext("Proportion Unique Taxa", side = 2, line = 3.5, cex = 1.5)
mtext("Habitat Type", side = 1, line = 3, cex = 1.5)
text(x = 3, y = 0.25, "*", cex = 2)
dev.off()
graphics.off()

t.test(1-water.frac, 1-sed.frac)
```

```{r fig.width=4, fig.height=4,echo=FALSE,fig.cap="Surface waters house more unique taxa than sediments do."}
img <- readPNG("../figures/Figure5.png")
grid.raster(img)
```

#### Figure 6: Surface Water vs. Sediment Communities
```{r}

png(filename = "../figures/Figure6.png",
    width = 2400, height = 1200, res = 96*2)
par(mfrow = c(1,2))

# Sediments Only PCoA
par(mar = c(5, 5, 3, 2) + 0.1)
plot(sediment.pcoa$points[ ,1], sediment.pcoa$points[ ,2], ylim = c(-0.4, 0.4), xlim = c(-.4,.4),
     xlab = paste("PCoA 1 (", s.var1, "%)", sep = ""),
     ylab = paste("PCoA 2 (", s.var2, "%)", sep = ""), 
     pch = 19, cex = 2.0, type = "n", cex.lab = 1.5, cex.axis = 1.2, axes = F)
axis(side = 1, labels = T, lwd.ticks = 2, cex.axis = 1.2, las = 1)
axis(side = 2, labels = T, lwd.ticks = 2, cex.axis = 1.2, las = 1)
axis(side = 3, labels = F, lwd.ticks = 2, cex.axis = 1.2, las = 1)
axis(side = 4, labels = F, lwd.ticks = 2, cex.axis = 1.2, las = 1)
abline(h = 0, v = 0, lty = 3)
box(lwd = 2)
mtext(side = 3, line = 2, "Sediments", cex = 1.5)
points(sediment.pcoa$points[which(sed.design$watershed == "LC"),1], 
       sediment.pcoa$points[which(sed.design$watershed == "LC"),2],
       pch=21, cex=2, bg="slategray2")
points(sediment.pcoa$points[which(sed.design$watershed == "WS01"),1], 
       sediment.pcoa$points[which(sed.design$watershed == "WS01"),2],
       pch=24, cex=2, bg="yellowgreen")
legend("topleft", c("Lookout Creek", "Watershed 01"),
       pt.bg = c("slategray2", "yellowgreen"), pch = c(21,24), cex = 1.5, bty = "n")

# Water Only PCoA
par(mar = c(5, 5, 3, 2) + 0.1)
plot(water.pcoa$points[ ,1], water.pcoa$points[ ,2], ylim = c(-0.4, 0.4), xlim = c(-.5,.4),
     xlab = paste("PCoA 1 (", w.var1, "%)", sep = ""),
     ylab = paste("PCoA 2 (", w.var2, "%)", sep = ""), 
     pch = 16, cex = 2.0, type = "n", cex.lab = 1.5, cex.axis = 1.2, axes = F)
axis(side = 1, labels = T, lwd.ticks = 2, cex.axis = 1.2, las = 1)
axis(side = 2, labels = T, lwd.ticks = 2, cex.axis = 1.2, las = 1)
axis(side = 3, labels = F, lwd.ticks = 2, cex.axis = 1.2, las = 1)
axis(side = 4, labels = F, lwd.ticks = 2, cex.axis = 1.2, las = 1)
abline(h = 0, v = 0, lty = 3)
box(lwd = 2)
mtext(side = 3, line = 2, "Water", cex = 1.5)
points(water.pcoa$points[which(water.design$watershed == "LC"),1], 
       water.pcoa$points[which(water.design$watershed == "LC"),2],
       pch=21, cex=2, bg="slategray2")
points(water.pcoa$points[which(water.design$watershed == "WS01"),1], 
       water.pcoa$points[which(water.design$watershed == "WS01"),2],
       pch=24, cex=2, bg="yellowgreen")
dev.off()
graphics.off()

```

```{r fig.width=4, fig.height=4,echo=FALSE,fig.cap="Ordination across the HJ Andrews Catchment"}
img <- readPNG("../figures/Figure6.png")
grid.raster(img)
```


### Organismal analysis
```{r}
top.taxa = matrix(nrow = nrow(OTUsREL), ncol = 5)
rownames(top.taxa) <- rownames(OTUsREL)
for(i in 1:nrow(OTUsREL)){
  top.taxa[i,] <- names(sort(tail(sort(OTUsREL[i,]),5), decreasing = T))
}


OTU.tax[OTU.tax$OTU %in% unique(as.vector(top.taxa[,5])),]

OTU.tax[OTU.tax$OTU %in% unique(as.vector(top.taxa[which(design$habitat == "water")])),]
OTU.tax[OTU.tax$OTU %in% unique(as.vector(top.taxa[which(design$habitat == "sediment")])),]

taxa.summary <- as.data.frame(cbind(rowSums(sediment.only > 1), rowSums(water.only > 1), 
                       rowSums(water.sed > 1), rowSums(in.sediment > 1),
                       rowSums(in.water > 1), rowSums(OTUs.PA)))
colnames(taxa.summary) <- c("sediments.only", "water.only", "shared", "in.sediment", "in.water", "total")

taxa.summary$water.only[which(design$habitat == "water")] / taxa.summary$total[which(design$habitat == "water")]
taxa.summary$sediments.only[which(design$habitat == "sediment")] / taxa.summary$total[which(design$habitat == "sediment")]
taxa.summary$shared / taxa.summary$total

# Unique to stream order
order.1.OTUs <- OTUs[, which(colSums(OTUs[which(design$order != "1"),]) == 0)]
order.2.OTUs <- OTUs[, which(colSums(OTUs[which(design$order != "2"),]) == 0)]
order.3.OTUs <- OTUs[, which(colSums(OTUs[which(design$order != "3"),]) == 0)]
order.4.OTUs <- OTUs[, which(colSums(OTUs[which(design$order != "4"),]) == 0)]
order.5.OTUs <- OTUs[, which(colSums(OTUs[which(design$order != "5"),]) == 0)]

unique(OTU.tax[OTU.tax$OTU %in% unique(as.vector(colnames(order.1.OTUs))),]$Order)
unique(OTU.tax[OTU.tax$OTU %in% unique(as.vector(colnames(order.2.OTUs))),]$Order)
unique(OTU.tax[OTU.tax$OTU %in% unique(as.vector(colnames(order.3.OTUs))),]$Order)
unique(OTU.tax[OTU.tax$OTU %in% unique(as.vector(colnames(order.4.OTUs))),]$Order)
unique(OTU.tax[OTU.tax$OTU %in% unique(as.vector(colnames(order.5.OTUs))),]$Order)

# site.by.phylum <- matrix(nrow = length(rownames(OTUsREL)), ncol = length(unique(OTU.tax[OTU.tax$OTU %in% colnames(OTUsREL),]$Phylum)))
# rownames(site.by.phylum) <- rownames(OTUsREL)
# colnames(site.by.phylum) <- unique(phyla.present)
# 
# # Taxonomic Analysis
# for(site in 1:length(rownames(OTUsREL))){
#   comm <- OTUsREL[site,]
#   taxa.present <- rownames(as.matrix(comm))
#   phyla.present <- (OTU.tax[OTU.tax$OTU %in% taxa.present,]$Phylum)
# 
#   
#   for(phylum in unique(phyla.present)){
#     site.by.phylum[site,phylum] <- sum(phyla.present == phylum) / length(phyla.present)
#   }
#   rownames(OTUsREL)[site]
# }
```

### Common Microbiomes
```{r}
sediment.common <- OTUsREL[, which(colSums(OTUs.PA[which(design$habitat == "sediment"),]) > 0.4*length(rownames(OTUs)))]
water.common <- OTUsREL[, which(colSums(OTUs.PA[which(design$habitat == "water"),]) > 0.4*length(rownames(OTUs)))]
intersect(colnames(sediment.common),colnames(water.common))

OTU.tax[OTU.tax$OTU %in% intersect(colnames(sediment.common),colnames(water.common)),]

```



Mantel Test
```{r, eval = F}

# Water Distances
water.lat <- as.numeric(env$latitude[which(design$habitat == "water")])
water.lon <- as.numeric(env$longitude[which(design$habitat == "water")])

water.struc.dist <- as.matrix(1 - vegdist(OTUsREL.log[which(design$habitat == "water"),]))
water.coord.dist <- as.matrix(dist(as.matrix(water.lat, water.lon)))

mantel(water.struc.dist, water.coord.dist, method = "pearson", permutations = 999)

# Sediment Distances
sed.lat <- as.numeric(env$latitude[which(design$habitat == "sediment")])
sed.lon <- as.numeric(env$longitude[which(design$habitat == "sediment")])

sed.struc.dist <- as.matrix(1 - vegdist(OTUsREL.log[which(design$habitat == "sediment"),]))
sed.coord.dist <- as.matrix(dist(as.matrix(sed.lat, sed.lon)))

mantel(sed.struc.dist, sed.coord.dist, method = "pearson", permutations = 999)


```


#### Figure S1: Surface-water / Sediment differences are robust across scales
```{r}

png(filename = "../figures/FigureS1.png",
    width = 2400, height = 1200, res = 96*2)
par(mfrow = c(1,2))

# Lookout Creek PCoA
par(mar = c(5, 5, 3, 2) + 0.1)
plot(lc.pcoa$points[ ,1], lc.pcoa$points[ ,2], ylim = c(-0.4, 0.35),
     xlab = paste("PCoA 1 (", lc.var1, "%)", sep = ""),
     ylab = paste("PCoA 2 (", lc.var2, "%)", sep = ""), 
     pch = 16, cex = 2.0, type = "n", cex.lab = 1.5, cex.axis = 1.2, axes = F)
axis(side = 1, labels = T, lwd.ticks = 2, cex.axis = 1.2, las = 1)
axis(side = 2, labels = T, lwd.ticks = 2, cex.axis = 1.2, las = 1)
abline(h = 0, v = 0, lty = 3)
box(lwd = 2)
mtext("Lookout Creek Watershed", side = 3, line = 2, cex = 1.5)
points(lc.pcoa$points[which(lc.design$habitat == "water"),1], 
       lc.pcoa$points[which(lc.design$habitat == "water"),2],
       pch=21, cex=2, bg="skyblue")
points(lc.pcoa$points[which(lc.design$habitat == "sediment"),1], 
       lc.pcoa$points[which(lc.design$habitat == "sediment"),2],
       pch=24, cex=2, bg="wheat")
legend("topleft", c("Sediment", "Water"),
         pt.bg = c("wheat", "skyblue"), pch = c(21,24), cex = 1.5, bty = "n")


# WS01 PCoA
par(mar = c(5, 5, 3, 2) + 0.1)
plot(w1.pcoa$points[ ,1], w1.pcoa$points[ ,2], ylim = c(-0.4, 0.35),
     xlab = paste("PCoA 1 (", w1.var1, "%)", sep = ""),
     ylab = paste("PCoA 2 (", w1.var2, "%)", sep = ""), 
     pch = 16, cex = 2.0, type = "n", cex.lab = 1.5, cex.axis = 1.2, axes = F)
axis(side = 1, labels = T, lwd.ticks = 2, cex.axis = 1.2, las = 1)
axis(side = 2, labels = T, lwd.ticks = 2, cex.axis = 1.2, las = 1)
abline(h = 0, v = 0, lty = 3)
box(lwd = 2)
mtext("Watershed 01", side = 3, line = 2, cex = 1.5)
points(w1.pcoa$points[which(w1.design$habitat == "water"),1], 
       w1.pcoa$points[which(w1.design$habitat == "water"),2],
       pch=21, cex=2, bg="skyblue")
points(w1.pcoa$points[which(w1.design$habitat == "sediment"),1], 
       w1.pcoa$points[which(w1.design$habitat == "sediment"),2],
       pch=24, cex=2, bg="wheat")

dev.off()
graphics.off()
```

```{r fig.width=4, fig.height=4,echo=FALSE,fig.cap="Differences between sediment and water communities exist on large and small spatial scales. "}
img <- readPNG("../figures/FigureS1.png")
grid.raster(img)
```

### CCA across watersheds and habitats
```{r, eval=FALSE}


# LC Watershed
lc.cca <- vegan::cca(OTUsREL.log[which(design$watershed == "LC"),] ~ env.mat[which(design$watershed == "LC"),]) 
#anova(lc.cca, by = "axis")
lc.cca.fit <- envfit(lc.cca, env.mat[which(design$watershed == "LC"),], perm = 999)
lc.cca.explainvar1 <- round(lc.cca$CCA$eig[1] /
                           sum(c(lc.cca$CCA$eig, lc.cca$CA$eig)), 3) * 100
lc.cca.explainvar2 <- round(lc.cca$CCA$eig[2] /
                           sum(c(lc.cca$CCA$eig, lc.cca$CA$eig)), 3) * 100

# WS01
w1.cca <- vegan::cca(OTUsREL.log[which(design$watershed == "WS01"),] ~ env.mat[which(design$watershed == "WS01"),]) 
#anova(w1.cca, by = "axis")
w1.cca.fit <- envfit(w1.cca, env.mat[which(design$watershed == "WS01"),], perm = 999)
w1.cca.explainvar1 <- round(w1.cca$CCA$eig[1] /
                           sum(c(w1.cca$CCA$eig, w1.cca$CA$eig)), 3) * 100
w1.cca.explainvar2 <- round(w1.cca$CCA$eig[2] /
                           sum(c(w1.cca$CCA$eig, w1.cca$CA$eig)), 3) * 100

# Water Only
water.cca <- vegan::cca(OTUsREL.log[which(design$habitat == "water"),] ~ env.mat[which(design$habitat == "water"),]) 
#anova(water.cca, by = "axis")
water.cca.fit <- envfit(water.cca, env.mat[which(design$habitat == "water"),], perm = 999)
water.cca.explainvar1 <- round(water.cca$CCA$eig[1] /
                           sum(c(water.cca$CCA$eig, water.cca$CA$eig)), 3) * 100
water.cca.explainvar2 <- round(water.cca$CCA$eig[2] /
                           sum(c(water.cca$CCA$eig, water.cca$CA$eig)), 3) * 100

# Sediments Only
sediment.cca <- vegan::cca(OTUsREL.log[which(design$habitat == "sediment"),] ~ env.mat[which(design$habitat == "sediment"),]) 
#anova(sediment.cca, by = "axis")
sediment.cca.fit <- envfit(sediment.cca, env.mat[which(design$habitat == "sediment"),], perm = 999)
sediment.cca.explainvar1 <- round(sediment.cca$CCA$eig[1] /
                           sum(c(sediment.cca$CCA$eig, sediment.cca$CA$eig)), 3) * 100
sediment.cca.explainvar2 <- round(sediment.cca$CCA$eig[2] /
                           sum(c(sediment.cca$CCA$eig, sediment.cca$CA$eig)), 3) * 100
```

#### Figure S2: Environmental variables explain little variation in structure. Elevation seems important.
```{r, eval=FALSE}
png(filename = "../figures/FigureS2.png",
    width = 2400, height = 1800, res=96*2)
par(mfrow=c(2,2))

# Lookout Plot
par(mar = c(5, 5, 4, 4) + 0.1)
plot(scores(lc.cca, display = "wa"), xlim = c(-2, 3), ylim = c(-2.5, 2),
     xlab = paste("CCA 1 (", lc.cca.explainvar1, "%)", sep = ""), 
     ylab = paste("CCA 2 (", lc.cca.explainvar2, "%)", sep = ""), 
     pch = 16, cex = 2.0, type = "n", cex.lab = 1.5, cex.axis = 1.2, axes = FALSE,
     main="Lookout Creek")
axis(side=1, labels=T, lwd.ticks=2, cex.axis=1.2, las=1)
axis(side=2, labels=T, lwd.ticks=2, cex.axis=1.2, las=1)
abline(h=0, v=0, lty=3)
box(lwd=2)
points(scores(lc.cca, display = "wa"),
       pch=19, cex=2, bg="gray", col="gray")
text(scores(lc.cca, display = "wa"), cex=0.5,
     labels = row.names(scores(lc.cca, display = "wa")))

lc.vectors <- scores(lc.cca, display = "bp")
row.names(lc.vectors) <- c("elev", "temp", "cond", "pH", "TN", "TP")
arrows(0, 0, lc.vectors[, 1] * 2, lc.vectors[, 2]*2,
       lwd = 2, lty = 1, length = 0.2, col = "red")
text(lc.vectors[, 1] * 2, lc.vectors[, 2] * 2, pos=3,
     labels = row.names(lc.vectors), col = "red")

# WS01 Plot
par(mar = c(5, 5, 4, 4) + 0.1)
plot(scores(w1.cca, display = "wa"), xlim = c(-1,3), ylim = c(-3, 2),
     xlab = paste("CCA 1 (", w1.cca.explainvar1, "%)", sep = ""), 
     ylab = paste("CCA 2 (", w1.cca.explainvar2, "%)", sep = ""), 
     pch = 16, cex = 2.0, type = "n", cex.lab = 1.5, cex.axis = 1.2, axes = FALSE,
     main = "Watershed 1")
axis(side=1, labels=T, lwd.ticks=2, cex.axis=1.2, las=1)
axis(side=2, labels=T, lwd.ticks=2, cex.axis=1.2, las=1)
abline(h=0, v=0, lty=3)
box(lwd=2)
points(scores(w1.cca, display = "wa"),
       pch=19, cex=2, bg="gray", col="gray")
text(scores(w1.cca, display = "wa"), cex = 0.5,
     labels = row.names(scores(w1.cca, display = "wa")))

w1.vectors <- scores(w1.cca, display = "bp")
row.names(w1.vectors) <- c("elev", "temp", "cond", "pH", "TN", "TP")
arrows(0, 0, w1.vectors[, 1] * 2, w1.vectors[, 2]*2,
       lwd = 2, lty = 1, length = 0.2, col = "red")
text(w1.vectors[, 1] * 2, w1.vectors[, 2] * 2, pos=3,
     labels = row.names(w1.vectors), col = "red")


#Water PLot
par(mar = c(5, 5, 4, 4) + 0.1)
plot(scores(water.cca, display = "wa"), xlim = c(-2, 2), ylim = c(-3, 2.5),
     xlab = paste("CCA 1 (", water.cca.explainvar1, "%)", sep = ""), 
     ylab = paste("CCA 2 (", water.cca.explainvar2, "%)", sep = ""), 
     pch = 16, cex = 2.0, type = "n", cex.lab = 1.5, cex.axis = 1.2, axes = FALSE,
     main = "Water")
axis(side=1, labels=T, lwd.ticks=2, cex.axis=1.2, las=1)
axis(side=2, labels=T, lwd.ticks=2, cex.axis=1.2, las=1)
abline(h=0, v=0, lty=3)
box(lwd=2)
points(scores(water.cca, display = "wa"),
       pch=19, cex=2, bg="gray", col="gray")
text(scores(water.cca, display = "wa"), cex = 0.5,
     labels = row.names(scores(water.cca, display = "wa")))

water.vectors <- scores(water.cca, display = "bp")
row.names(water.vectors) <- c("elev", "temp", "cond", "pH", "TN", "TP")
arrows(0, 0, water.vectors[, 1] * 2, water.vectors[, 2]*2,
       lwd = 2, lty = 1, length = 0.2, col = "red")
text(water.vectors[, 1] * 2, water.vectors[, 2] * 2, pos=3,
     labels = row.names(water.vectors), col = "red")

# Sed plot

par(mar = c(5, 5, 4, 4) + 0.1)
plot(scores(sediment.cca, display = "wa"), xlim = c(-2, 3), ylim = c(-1.25, 1.25),
     xlab = paste("CCA 1 (", sediment.cca.explainvar1, "%)", sep = ""), 
     ylab = paste("CCA 2 (", sediment.cca.explainvar2, "%)", sep = ""), 
     pch = 16, cex = 2.0, type = "n", cex.lab = 1.5, cex.axis = 1.2, axes = FALSE,
     main = "Sediment")
axis(side=1, labels=T, lwd.ticks=2, cex.axis=1.2, las=1)
axis(side=2, labels=T, lwd.ticks=2, cex.axis=1.2, las=1)
abline(h=0, v=0, lty=3)
box(lwd=2)
points(scores(sediment.cca, display = "wa"),
       pch=19, cex=2, bg="gray", col="gray")
text(scores(sediment.cca, display = "wa"), cex = 0.5,
     labels = row.names(scores(sediment.cca, display = "wa")))

sediment.vectors <- scores(sediment.cca, display = "bp")
row.names(sediment.vectors) <- c("elev", "temp", "cond", "pH", "TN", "TP")
arrows(0, 0, sediment.vectors[, 1] * 2, sediment.vectors[, 2]*2,
       lwd = 2, lty = 1, length = 0.2, col = "red")
text(sediment.vectors[, 1] * 2, sediment.vectors[, 2] * 2, pos=3,
     labels = row.names(sediment.vectors), col = "red")

dev.off()
graphics.off()
```

```{r fig.width=4, fig.height=4,echo=FALSE,fig.cap="Constrained ordination of sites"}
img <- readPNG("../figures/FigureS2.png")
grid.raster(img)
```


```{r}
hja.bray <- 1 - (vegdist(OTUsREL.log, method = "bray"))
hja.bray.list <- liste(hja.bray, entry = "struc")
hja.elev.dist <- dist(env[,10])
hja.elev.dist.list <- liste(hja.elev.dist, entry = "elev")
hja.model <- lm(hja.bray.list$struc ~ hja.elev.dist.list$elev)
summary(hja.model)

png(filename = "../figures/FigureS5.png", 
    width = 1200, height = 1200, res = 96*2)
par(mar = c(5, 5, 1, 3) + 0.1)
plot(hja.bray.list$struc ~ hja.elev.dist.list$elev, 
     xlim = c(0, 800), xlab="",
     ylab = "", xaxt = "n", yaxt = "n")
axis(side=1, labels=T, lwd.ticks=2, cex.axis=1.2, las=1)
axis(side=2, labels=T, lwd.ticks=2, cex.axis=1.2, las=1)
axis(side=3, labels=F, lwd.ticks=2, cex.axis=1.2, las=1)
axis(side=4, labels=F, lwd.ticks=2, cex.axis=1.2, las=1)
mtext("Bray-Curtis Similarity", side = 2, line = 3, cex = 1.2)
mtext("Difference in Elevation (m)", side = 1, line = 3, cex = 1.5)
box(lwd=2)
abline(hja.model, lwd = 2)
dev.off()
graphics.off()
img <- readPNG("../figures/FigureS5.png")
grid.raster(img)
```

# Elevation gradient analysis
```{r}
water.bray <- 1 - (vegdist(OTUsREL.log[which(design$habitat=="water"),], method = "bray"))
water.bray.list <- liste(water.bray, entry = "struc")
water.elev.dist <- dist(env[which(design$habitat == "water"),10])
water.elev.dist.list <- liste(water.elev.dist, entry = "elev")
model.water <- lm(water.bray.list$struc ~ water.elev.dist.list$elev)
summary(model.water)

sed.bray <- 1 - (vegdist(OTUsREL.log[which(design$habitat=="sediment"),], method = "bray"))
sed.bray.list <- liste(sed.bray, entry = "struc")
sed.elev.dist <- dist(env[which(design$habitat == "sediment"),10])
sed.elev.dist.list <- liste(sed.elev.dist, entry = "elev")
model.sed <- lm(sed.bray.list$struc ~ sed.elev.dist.list$elev)
summary(model.sed)
```

### Figure S3: Bray-Curtis Similarity by Elevation
```{r}
png(filename = "../figures/FigureS3.png", 
    width = 1200, height = 1800, res = 96*2)
par(mfrow = c(2, 1))
par(mar = c(1, 5, 4, 3) + 0.1)

plot(water.bray.list$struc ~ water.elev.dist.list$elev, 
     xlim = c(0, 800), ylim = c(.1, .6), xlab="",
     ylab = "", xaxt = "n", yaxt = "n")
axis(side=1, labels=F, lwd.ticks=2, cex.axis=1.2, las=1)
axis(side=2, labels=T, lwd.ticks=2, cex.axis=1.2, las=1)
axis(side=3, labels=F, lwd.ticks=2, cex.axis=1.2, las=1)
axis(side=4, labels=F, lwd.ticks=2, cex.axis=1.2, las=1)
mtext("Surface Water", side = 2, line = 3, cex = 1.2)
mtext("Bray-Curtis Similarity", side=3, line=1, cex=1.5)
box(lwd=2)

par(mar = c(5, 5, 1, 3) + 0.1)


plot(sed.bray.list$struc ~ sed.elev.dist.list$elev,
     xlim = c(0,800), ylim = c(0.1,0.6), xlab="",
     ylab = "", xaxt = "n", yaxt = "n")
axis(side=1, labels=T, lwd.ticks=2, cex.axis=1.2, las=1)
axis(side=2, labels=T, lwd.ticks=2, cex.axis=1.2, las=1)
axis(side=3, labels=F, lwd.ticks=2, cex.axis=1.2, las=1)
axis(side=4, labels=F, lwd.ticks=2, cex.axis=1.2, las=1)
mtext("Sediment", side = 2, line = 3, cex = 1.2)
mtext("Elevation difference (m)", side = 1, line = 3, cex = 1.5)
box(lwd=2)
abline(model.sed, lwd = 2)


dev.off()
graphics.off()

```

```{r fig.width=4, fig.height=4,echo=FALSE,fig.cap="Beta-diversity in surface waters and sediments by elevation."}
img <- readPNG("../figures/FigureS3.png")
grid.raster(img)
```

### Calculate Dendritic Distances
```{r}
require('igraph')
adj.mat <- as.matrix(read.csv("../data/undirected-matrix.csv", header=T))
row.names(adj.mat) <- adj.mat[,1]
adj.mat <- adj.mat[,-1]
adj.mat <- (adj.mat == 1) * 1

stream.network <- graph_from_adjacency_matrix(adjmatrix = adj.mat)

png(filename = "../figures/stream-network.png", 
    height = 4800, width = 4800, res = 2*96)
plot.igraph(stream.network)
dev.off()

 # Create Dist Matrix
xy.total <- cbind(env.total$longitude, env.total$latitude)
xy.total <- project(xy.total, "+proj=utm +zone=10 +ellps=WGS84")
euc.dist.mat <- as.matrix(dist(xy.total, method = "euclidean"))
rownames(euc.dist.mat) <- env.total$sample
colnames(euc.dist.mat) <- env.total$sample

# Function to find paths along the network.
dend.dist <- function(graph = "", dist.mat = ""){

  dend.dist.mat <- matrix(data = NA, nrow = length(env.total$sample), 
                          ncol = length(env.total$sample))
  rownames(dend.dist.mat) <- env.total$sample
  colnames(dend.dist.mat) <- env.total$sample
  
  for(row in rownames(dend.dist.mat)){
    for(col in colnames(dend.dist.mat)){
      path.dist <- 0
      path <- shortest_paths(graph = graph, from = row, to = col)$vpath[[1]]
      
      if(length(path) > 1){
        for(i in 1:(length(path) - 1)){
          path.dist <- path.dist + dist.mat[path[[i]]$name, path[[i+1]]$name]
        }
      }
      
      dend.dist.mat[row,col] <- path.dist
    }
  }
  
  return(dend.dist.mat)
}

dend.dist.mat <- dend.dist(graph = stream.network, dist.mat = euc.dist.mat)

net = graph.adjacency((dend.dist.mat * adj.mat), mode = "directed",
                      weighted = TRUE)
plot.igraph(net)
```


### Figure S4: Distance-Decay
```{r, eval = FALSE}
xy <- cbind(env$longitude, env$latitude)
geo.dists <- geoXY(env$latitude, env$longitude)
xy <- project(xy, "+proj=utm +zone=10 +ellps=WGS84")
dist.mat <- as.matrix(dist(xy, method = "euclidean"))
dists <- dend.dist.mat[which(rownames(dend.dist.mat) %in% rownames(design)),
                       which(rownames(dend.dist.mat) %in% rownames(design))]
dists[!lower.tri(dists)] <- NA 

# Water Distance Decay
#water.lat <- as.numeric(env$latitude[which(design$habitat == "water")])
#water.lon <- as.numeric(env$longitude[which(design$habitat == "water")])
water.xy <- xy[which(design$habitat == "water"),]
water.coord.dist <- dists[which(design$habitat == "water"),which(design$habitat == "water")]
water.struc.dist <- 1 - vegdist(OTUsREL.log[which(design$habitat == "water"),])
#water.coord.dist <- dist(as.matrix(water.lat, water.lon))
#water.env.dist <- 1 - vegdist(env.mat[which(design$habitat == "water"),])
water.struc.dist.ls <- liste(water.struc.dist, entry = "struc")
#water.env.dist.ls <- liste(water.env.dist, entry = "env")
water.coord.dist.ls <- liste(water.coord.dist, entry = "dist")
water.distances <- data.frame(na.omit(water.coord.dist.ls), water.struc.dist.ls[,3])
names(water.distances)[4] <- c("struc")

# Sediment Distance Decay
#sed.lat <- as.numeric(env$latitude[which(design$habitat == "sediment")])
#sed.lon <- as.numeric(env$longitude[which(design$habitat == "sediment")])
sed.xy <- xy[which(design$habitat == "sediment"),]
sed.coord.dist <- dists[which(design$habitat == "sediment"),which(design$habitat == "sediment")]
sed.struc.dist <- 1 - vegdist(OTUsREL.log[which(design$habitat == "sediment"),])
#sed.coord.dist <- dist(as.matrix(sed.lat, sed.lon))
#sed.env.dist <- 1 - vegdist(env.mat[which(design$habitat == "sediment"),])
sed.struc.dist.ls <- liste(sed.struc.dist, entry = "struc")
#sed.env.dist.ls <- liste(sed.env.dist, entry = "env")
sed.coord.dist.ls <- liste(sed.coord.dist, entry = "dist")
sed.distances <- data.frame(na.omit(sed.coord.dist.ls), sed.struc.dist.ls[,3])
names(sed.distances)[4] <- c("struc")

## Make Figure
png(filename = "../figures/FigureS4.png",
    width = 1200, height = 1200, res = 96*2)
par(mfrow = c(2, 1) + 0.5)

par(mar = c(1, 5, 3, 3))
plot(water.distances$dist, water.distances$struc, xlab="", 
     ylab = "", xaxt="n", yaxt="n", 
     )
(water.lm <- (lm(water.distances$struc ~ water.distances$dist)))
abline(water.lm, lwd = 2)
axis(side=1, labels=F, lwd.ticks=2, cex.axis=1.2, las=1)
axis(side=2, labels=T, lwd.ticks=2, cex.axis=1.2, las=1)
axis(side=3, labels=F, lwd.ticks=2, cex.axis=1.2, las=1)
axis(side=4, labels=F, lwd.ticks=2, cex.axis=1.2, las=1)
box(lwd = 2)
mtext("Surface Water", side = 2, line = 3, cex = 1.2)

par(mar = c(4, 5, 1, 3))
plot(sed.distances$dist, sed.distances$struc, xlab="", 
     ylab = "", xaxt = "n", yaxt = "n", )
axis(side=1, labels=T, lwd.ticks=2, cex.axis=1.2, las=1)
axis(side=2, labels=T, lwd.ticks=2, cex.axis=1.2, las=1)
axis(side=3, labels=F, lwd.ticks=2, cex.axis=1.2, las=1)
axis(side=4, labels=F, lwd.ticks=2, cex.axis=1.2, las=1)
box(lwd=2)
mtext("Sediment", side = 2, line = 3, cex = 1.2)
(sed.lm <- (lm(sed.distances$struc ~ sed.distances$dist)))
abline(sed.lm, lwd = 2)

mtext("Dendritic distance (m)", side = 1, line = 3, cex = 1.5)

dev.off()
graphics.off()

diffslope(sed.distances$dist, sed.distances$struc, water.distances$dist, water.distances$struc)
```

```{r fig.width=4, fig.height=4,echo=FALSE,fig.cap="Distance-decay relationship."}
img <- readPNG("../figures/FigureS4.png")
grid.raster(img)
```

### GIS and Imagery
```{r, eval=FALSE}
DEM <- raster("~/GitHub/HJA-streams/imagery/bare_earth_DEM_1m/gi01001.e00")
DEM
image(DEM, col=terrain.colors(1000))
streams <- readOGR(dsn = "./imagery/stream_network_2008/", layer = "lidar_stream")
streams@data
plot(streams, add = TRUE)
summary(streams)
```

### Variance Partitioning

#### Var Part - HJA Catchment
```{r}
trunc.dist <- as.matrix(dist(geo.dists)) * adj.mat
dist.pcnm <- pcnm(trunc.dist)
ordisurf(geo.dists, scores(dist.pcnm, choi=1), bubble = 4, main = "PCNM 1")
ordisurf(geo.dists, scores(dist.pcnm, choi=2), bubble = 4, main = "PCNM 2")
ordisurf(geo.dists, scores(dist.pcnm, choi=3), bubble = 4, main = "PCNM 3")
ordisplom(dist.pcnm, choices = 1:4)
rs <- rowSums(OTUsREL.log) / sum(OTUsREL.log)
pcnmw <- pcnm(trunc.dist, w = rs)

spatial.var <- vegan::cca(OTUsREL.log ~ scores(pcnmw))
env.var <- vegan::cca(OTUsREL.log ~ env.mat[,c(1:4,6)])
env.spat.var <- vegan::cca(OTUsREL.log ~ scores(pcnmw) +
                             Condition(env.mat[,c(1:4,6)]))
spat.env.var <- vegan::cca(OTUsREL.log ~ env.mat[,c(1:4,6)] +
                             Condition(scores(pcnmw)))



sum(env.var$CCA$eig)
sum(spatial.var$CCA$eig)
sum(env.spat.var$CCA$eig)
sum(spat.env.var$CCA$eig)

hja.ca <- vegan::cca(OTUsREL.log)
sum(hja.ca$CA$eig)

var.part.1 <- sum(env.var$CCA$eig) / sum(hja.ca$CA$eig)
var.part.2 <- sum(spatial.var$CCA$eig) / sum(hja.ca$CA$eig)
var.part.3 <- sum(spat.env.var$CCA$eig) / sum(hja.ca$CA$eig)
var.part.4 <- sum(env.spat.var$CCA$eig) / sum(hja.ca$CA$eig)

print(paste("HJA Pure Env. = ", var.part.1))
print(paste("HJA Pure Space. = ", var.part.2))
print(paste("HJA Spatial + Env = ", (var.part.1 - var.part.3)))
print(paste("HJA Residual Var. = ", (1- sum(var.part.1 + var.part.2))))
```

#### Var Part - Surface Water
```{r}
env.mat <- env.mat[,c(1:4,6)]
water.dist.pcnm <- pcnm(dist(xy[which(design$habitat == "water"),]))
water.rs <- rowSums(OTUsREL.log[which(design$habitat == "water"),]) / 
  sum(OTUsREL.log[which(design$habitat == "water"),])
water.pcnmw <- pcnm(dist(xy[which(design$habitat == "water"),]), w = water.rs)

water.spatial.var <- vegan::cca(OTUsREL.log[which(design$habitat == "water"),] ~ scores(water.pcnmw))
water.env.var <- vegan::cca(OTUsREL.log[which(design$habitat == "water"),] ~ env.mat[which(design$habitat == "water"),])
water.env.spat.var <- vegan::cca(OTUsREL.log[which(design$habitat == "water"),] ~
      scores(water.pcnmw) + Condition(env.mat[which(design$habitat == "water"),]))
water.spat.env.var <- vegan::cca(OTUsREL.log[which(design$habitat == "water"),] ~ 
      env.mat[which(design$habitat == "water"),] + Condition(scores(water.pcnmw)))

sum(water.env.var$CCA$eig)
sum(water.spatial.var$CCA$eig)
sum(water.env.spat.var$CCA$eig)
sum(water.spat.env.var$CCA$eig)

water.ca <- vegan::cca(OTUsREL.log[which(design$habitat == "water"),])
sum(water.ca$CA$eig)

water.var.part.1 <- sum(water.env.var$CCA$eig) / sum(water.ca$CA$eig)
water.var.part.2 <- sum(water.spatial.var$CCA$eig) / sum(water.ca$CA$eig)
water.var.part.3 <- sum(water.spat.env.var$CCA$eig) / sum(water.ca$CA$eig)
water.var.part.4 <- sum(water.env.spat.var$CCA$eig) / sum(water.ca$CA$eig)

print(paste("Water Pure Env. = ", water.var.part.1))
print(paste("Water Pure Space. = ", water.var.part.2))
print(paste("Water Spatial + Env = ", (water.var.part.1 - water.var.part.3)))
print(paste("Water Residual Var. = ", (1- sum(water.var.part.1 + water.var.part.2))))
```

#### Var Part - Sediments Only
```{r}
sediment.dist.pcnm <- pcnm(dist(xy[which(design$habitat == "sediment"),]))
sediment.rs <- rowSums(OTUsREL.log[which(design$habitat == "sediment"),]) / 
  sum(OTUsREL.log[which(design$habitat == "sediment"),])
sediment.pcnmw <- pcnm(dist(xy[which(design$habitat == "sediment"),]), w = sediment.rs)

sediment.spatial.var <- vegan::cca(OTUsREL.log[which(design$habitat == "sediment"),] ~ scores(sediment.pcnmw))
sediment.env.var <- vegan::cca(OTUsREL.log[which(design$habitat == "sediment"),] ~ env.mat[which(design$habitat == "sediment"),])
sediment.env.spat.var <- vegan::cca(OTUsREL.log[which(design$habitat == "sediment"),] ~
      scores(sediment.pcnmw) + Condition(env.mat[which(design$habitat == "sediment"),]))
sediment.spat.env.var <- vegan::cca(OTUsREL.log[which(design$habitat == "sediment"),] ~ 
      env.mat[which(design$habitat == "sediment"),] + Condition(scores(sediment.pcnmw)))

sum(sediment.env.var$CCA$eig)
sum(sediment.spatial.var$CCA$eig)
sum(sediment.env.spat.var$CCA$eig)
sum(sediment.spat.env.var$CCA$eig)

sediment.ca <- vegan::cca(OTUsREL.log[which(design$habitat == "sediment"),])
sum(sediment.ca$CA$eig)

sediment.var.part.1 <- sum(sediment.env.var$CCA$eig) / sum(sediment.ca$CA$eig)
sediment.var.part.2 <- sum(sediment.spatial.var$CCA$eig) / sum(sediment.ca$CA$eig)
sediment.var.part.3 <- sum(sediment.spat.env.var$CCA$eig) / sum(sediment.ca$CA$eig)
sediment.var.part.4 <- sum(sediment.env.spat.var$CCA$eig) / sum(sediment.ca$CA$eig)

print(paste("Sediment Pure Env. = ", sediment.var.part.1))
print(paste("Sediment Pure Space. = ", sediment.var.part.2))
print(paste("Sediment Spatial + Env = ", (sediment.var.part.1 - sediment.var.part.3)))
print(paste("Sediment Residual Var. = ", (1- sum(sediment.var.part.1 + sediment.var.part.2))))
```

```{r}
habitat <- (env[,8] == "sediment") * 1
env.dummy <- as.matrix(cbind(habitat, env[,c(10, 11, 12, 13, 15)]))

d.spatial.var <- spatial.var
d.env.var <- vegan::cca(OTUsREL.log ~ env.dummy)
d.env.spat.var <- vegan::cca(OTUsREL.log ~ scores(pcnmw) +
                             Condition(env.dummy))
d.spat.env.var <- vegan::cca(OTUsREL.log ~ env.dummy +
                             Condition(scores(pcnmw)))

sum(d.env.var$CCA$eig)
sum(d.spatial.var$CCA$eig)
sum(d.env.spat.var$CCA$eig)
sum(d.spat.env.var$CCA$eig)

d.var.part.1 <- sum(d.env.var$CCA$eig) / sum(hja.ca$CA$eig)
d.var.part.2 <- sum(d.spatial.var$CCA$eig) / sum(hja.ca$CA$eig)
d.var.part.3 <- sum(d.spat.env.var$CCA$eig) / sum(hja.ca$CA$eig)
d.var.part.4 <- sum(d.env.spat.var$CCA$eig) / sum(hja.ca$CA$eig)
```



```{r}
# sediments
hja.pa.db <- vegdist(OTUs.PA[which(design$habitat == "sediment"),], method = "bray", upper = TRUE, diag = TRUE)
hja.pa.pcoa <- cmdscale(hja.pa.db, eig=TRUE, k=3)
var1.pa <- round(hja.pa.pcoa$eig[1] / sum(hja.pa.pcoa$eig),3) * 100
var2.pa <- round(hja.pa.pcoa$eig[2] / sum(hja.pa.pcoa$eig),3) * 100
var3.pa <- round(hja.pa.pcoa$eig[3] / sum(hja.pa.pcoa$eig),3) * 100

par(mar = c(5, 5, 3, 2) + 0.1)
plot(hja.pa.pcoa$points[ ,1], hja.pa.pcoa$points[ ,2],
     xlab = paste("PCoA 1 (", var1.pa, "%)", sep = ""),
     ylab = paste("PCoA 2 (", var2.pa, "%)", sep = ""), 
     pch = 19, cex = 2.0, type = "n", cex.lab = 1.5, cex.axis = 1.2, axes = F,
     main="sediment")
axis(side = 1, labels = T, lwd.ticks = 2, cex.axis = 1.2, las = 1)
axis(side = 2, labels = T, lwd.ticks = 2, cex.axis = 1.2, las = 1)
axis(side = 3, labels = F, lwd.ticks = 2, cex.axis = 1.2, las = 1)
axis(side = 4, labels = F, lwd.ticks = 2, cex.axis = 1.2, las = 1)
abline(h = 0, v = 0, lty = 3)
box(lwd = 2)
points(hja.pa.pcoa$points[which(sed.design$watershed == "LC"),1], 
       hja.pa.pcoa$points[which(sed.design$watershed == "LC"),2],
       pch=19, cex=2, bg="wheat", col="wheat")
points(hja.pa.pcoa$points[which(sed.design$watershed == "WS01"),1], 
       hja.pa.pcoa$points[which(sed.design$watershed == "WS01"),2],
       pch=17, cex=2, bg="skyblue", col="skyblue")
text(hja.pa.pcoa$points[ ,1], hja.pa.pcoa$points[ ,2],
      cex = 0.5, labels = row.names(hja.pa.pcoa$points))
legend("topleft", c("lookout", "ws01"), col = c("wheat", "skyblue"),
         pt.bg = c("wheat", "skyblue"), pch = c(19,17), cex = 1.5, bty = "n")

# water
hja.pa.db <- vegdist(OTUs.PA[which(design$habitat == "water"),], method = "bray", upper = TRUE, diag = TRUE)
hja.pa.pcoa <- cmdscale(hja.pa.db, eig=TRUE, k=3)
var1.pa <- round(hja.pa.pcoa$eig[1] / sum(hja.pa.pcoa$eig),3) * 100
var2.pa <- round(hja.pa.pcoa$eig[2] / sum(hja.pa.pcoa$eig),3) * 100
var3.pa <- round(hja.pa.pcoa$eig[3] / sum(hja.pa.pcoa$eig),3) * 100

par(mar = c(5, 5, 3, 2) + 0.1)
plot(hja.pa.pcoa$points[ ,1], hja.pa.pcoa$points[ ,2],
     xlab = paste("PCoA 1 (", var1.pa, "%)", sep = ""),
     ylab = paste("PCoA 2 (", var2.pa, "%)", sep = ""), 
     pch = 19, cex = 2.0, type = "n", cex.lab = 1.5, cex.axis = 1.2, axes = F,
     main="water")
axis(side = 1, labels = T, lwd.ticks = 2, cex.axis = 1.2, las = 1)
axis(side = 2, labels = T, lwd.ticks = 2, cex.axis = 1.2, las = 1)
axis(side = 3, labels = F, lwd.ticks = 2, cex.axis = 1.2, las = 1)
axis(side = 4, labels = F, lwd.ticks = 2, cex.axis = 1.2, las = 1)
abline(h = 0, v = 0, lty = 3)
box(lwd = 2)
points(hja.pa.pcoa$points[which(water.design$watershed == "LC"),1], 
       hja.pa.pcoa$points[which(water.design$watershed == "LC"),2],
       pch=19, cex=2, bg="wheat", col="wheat")
points(hja.pa.pcoa$points[which(water.design$watershed == "WS01"),1], 
       hja.pa.pcoa$points[which(water.design$watershed == "WS01"),2],
       pch=17, cex=2, bg="skyblue", col="skyblue")
text(hja.pa.pcoa$points[ ,1], hja.pa.pcoa$points[ ,2],
      cex = 0.5, labels = row.names(hja.pa.pcoa$points))
legend("topleft", c("lookout", "ws01"), col = c("wheat", "skyblue"),
         pt.bg = c("wheat", "skyblue"), pch = c(19,17), cex = 1.5, bty = "n")


```

# heatmaps and clustering
```{r}
jet.colors <- colorRampPalette(c("#00007F", "blue", "#007FFF", "cyan", 
                                 "#7FFF7F", "yellow", "#FF7F00", "red", 
                                 "#7F0000"))
order <- c(design$site[design$habitat == "water"], design$site[design$habitat == "sediment"])
levelplot(as.matrix(hja.db)[order,order], aspect = "iso", col.regions = jet.colors,
          xlab = "site", ylab = "site", scales = list(cex = 0.5),
          main = "Bray-Curtis Distance")

hja.ward <- hclust(hja.db, method = "ward.D2")
par(mar = c(1,5,2,2) + 0.1)
plot(hja.ward)

require(gplots)
heatmap.2(as.matrix(hja.db), distfun = function(x) vegdist(x, method = "bray"),
          hclustfun = function(x) hclust(x, method = "ward.D2"),
          col = jet.colors(100), trace = "none", density.info = "none")
```

```{r}
hja.pca <- princomp(env.mat)
summary(hja.pca)
plot(hja.pca, type = "l")
biplot(hja.pca)
pc1 <- hja.pca$scores[,1]
pc2 <- hja.pca$scores[,2]

env.dists <- liste(dist(pc1), entry = "env")[,3]
hja.dists <- cbind(liste(hja.db, entry = "comm.struc"), env.dists)
plot(hja.dists$env.dists, hja.dists$comm.struc)

summary(lm(hja.dists$comm.struc ~ hja.dists$env.dists))

water.pca <- princomp(env.mat[which(design$habitat == "water"),])
summary(water.pca)
water.pc1 <- water.pca$scores[,1]
water.env.dists <- liste(dist(water.pc1), entry = "env")[,3]
water.dists <- cbind(liste(water.db, entry = "comm.struc"), water.env.dists)
plot(water.dists$water.env.dists, water.dists$comm.struc)

sed.pca <- princomp(env.mat[which(design$habitat == "sediment"),])
summary(sed.pca)
sed.pc1 <- sed.pca$scores[,1]
sed.env.dists <- liste(dist(sed.pc1), entry = "env")[,3]
sed.dists <- cbind(liste(sediment.db, entry = "comm.struc"), sed.env.dists)
plot(sed.dists$sed.env.dists, sed.dists$comm.struc)

summary(lm(sed.dists$comm.struc ~ sed.dists$sed.env.dists))
summary(lm(water.dists$comm.struc ~ water.dists$water.env.dists))
```

# Community-Environmental similarity 
```{r}
png(filename = "../figures/FigureS6.png",
    width = 1200, height = 1200, res = 96*2)
par(mfrow = c(2, 1) + 0.5)

par(mar = c(1, 5, 3, 3) + 0.4)
plot(water.dists$water.env.dists, water.dists$comm.struc, xlab="", 
     ylab = "", xaxt="n", yaxt="n", 
     xlim = c(-.15, 6.5), ylim = c(0.4,.9))
abline(lm(water.dists$comm.struc ~ water.dists$water.env.dists), lty = 2, lwd = 2)
axis(side=1, labels=F, lwd.ticks=2, cex.axis=1.2, las=1)
axis(side=2, labels=T, lwd.ticks=2, cex.axis=1.2, las=1)
axis(side=3, labels=F, lwd.ticks=2, cex.axis=1.2, las=1)
axis(side=4, labels=F, lwd.ticks=2, cex.axis=1.2, las=1)
box(lwd = 2)
mtext("Community Dissimilarity\nSurface Water", side = 2, line = 3, cex = 1.2)

par(mar = c(4, 5, 1, 3) + 0.4)
plot(sed.dists$sed.env.dists, sed.dists$comm.struc, xlab="", 
     ylab = "", xaxt = "n", yaxt = "n", 
     xlim = c(-.15, 6.5), ylim = c(0.4,.9))
axis(side=1, labels=T, lwd.ticks=2, cex.axis=1.2, las=1)
axis(side=2, labels=T, lwd.ticks=2, cex.axis=1.2, las=1)
axis(side=3, labels=F, lwd.ticks=2, cex.axis=1.2, las=1)
axis(side=4, labels=F, lwd.ticks=2, cex.axis=1.2, las=1)
box(lwd=2)
mtext("Community Dissimilarity\nSediment", side = 2, line = 3, cex = 1.2)
abline(lm(sed.dists$comm.struc ~ sed.dists$sed.env.dists), lwd = 2)

mtext("Environmental Distance (PC1)", side = 1, line = 3, cex = 1.5)


#mtext("Community Similarity", side = 2, line = 3, cex = 1.5)

dev.off()
graphics.off()

#diffslope(sed.distances$dist, sed.distances$struc, water.distances$dist, water.distances$struc)


```

```{r fig.width=4, fig.height=4,echo=FALSE,fig.cap="Beta-diversity in surface waters and sediments by elevation."}
img <- readPNG("../figures/FigureS6.png")
grid.raster(img)
```

# Co-occurrence analysis
```{R}
require(cooccur)
hja.cooccur <- cooccur(mat = as.data.frame(OTUs.PA),
                       type = "site_spp",
                       spp_names = T,
                       thresh = T)
```

# Rank-Abundance Curve
```{r}
RAC <- function(x = ""){
  x = as.vector(x)
  x.ab = x[x > 0]
  x.ab.ranked = x.ab[order(x.ab, decreasing = TRUE)]
  return(x.ab.ranked)
  }

sed.sites <- OTUsREL.log[which(design$habitat == "sediment"),]
water.sites <- OTUsREL.log[which(design$habitat == "water"),]
rac <- RAC(sed.sites[1,])
ranks <- as.vector(seq(1, length(rac)))
par(mar=c(5.1, 5.1, 4.1, 2.1))                        
plot(ranks, log(rac), type = 'l', axes=F, col = "brown",           
     xlab = "Rank in abundance", ylab = "Abundance",
     las = 1, cex.lab = 1.4, cex.axis = 1.25,
     ylim = c(0, log(20)), xlim = c(0, max(alpha.div$S.obs)))
for(site in 2:nrow(sed.sites)){
  rac <- RAC(sed.sites[site,])
  ranks <- as.vector(seq(1, length(rac)))
  points(ranks, log(rac), type = 'l', col = "brown")
}
for(site in 1:nrow(water.sites)){
  rac <- RAC(water.sites[site,])
  ranks <- as.vector(seq(1, length(rac)))
  points(ranks, log(rac), type = 'l', col = "blue")
}

box()                                                 
axis(side = 1, labels=T, cex.axis = 1.25)             
axis(side = 2, las = 1, cex.axis = 1.25,              
     labels=c(1, 2, 5, 10, 20), at=log(c(1, 2, 5, 10, 20)))

```
			   